---
title: "Plankton response to salinity (aka 2015 CSI Dispersal Experiment)"
author: "Jo Werba, Alex Stucy, Mike McCoy, Ariane Peralta with contributions from Mario Muscarella (phylogenetic analyses, data manipulation, mothur tools)"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document: null
  fig_caption: yes
  html_document: default
editor_options:
  chunk_output_type: console
---
Project Description: Analysis of salinity and dispersal of saline aquatic communities on bacterial community structure and function. Manuscript preparation for zooplankton/bacterioplankton community composition-function study.

# Setting up working directory, packages
```{r Setup, include=FALSE}
#PC - set WD manually by Session -> Set Working Directory -> Choose Directory...
if(Sys.info()[1] == "Darwin"){
  setwd("~/GitHub/CSI_Dispersal/analyses/")
} else {
  # setwd(choose.dir())
}
rm(list = ls())

#Set source R tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

#load req'd packages
require("vegan")
require("dplyr")
require("nlme")
require("reshape2")
require("ecodist")
require("ggplot2")
require("ade4")
require("png")
require("MASS")
require("grid")
require("ape")
require("picante")
# require("venneuler")
require("Hmisc")
require("knitr")
# set std err
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}
ci <- function(x, ...) {
1.96 * sd(x, na.rm = TRUE)
}
```

#Loading data file
```{r Loading Files, include=FALSE}
#NOTE: Tank 3 only had fresh water added. Tank 1 was the source for the salt and tank 2 was the source for the fresh water.
#NOTE: Dispersal 0 (treatments 1 and 2) were source tanks-- no dispersal. Dispersal 1 (treatment 3) only received fresh water. Dispersal 3 (treatments 4,6,8,10) only got salt water. And dispersal 2 (treatments 5,7,9,11) received both fresh and salt.

# load design file - no source tanks
design.ns <- read.csv("../data/CSI_Design_ENV_NoSourceTanks.csv", row.names=1)
#head(design.ns)
#str(design.ns)
dim(design.ns)

# load design file - with source tanks but missing decomp, nutrients, sreal, Date2 columns
design.full <- read.csv("../data/design_CSI.csv", row.names=1)
design.full <- design.full[-c(grep("mock community", design.full$CSI_ID)), ]
dim(design.full)

# load OTU file
#csi_otu <- read.otu("../data/CSI.shared")
csi_otu <- read.otu("../data/CSIvsearch.shared")
dim(csi_otu)

# load environ/function file - includes expt design, decomp data, Cmin, nutrients day 45 only - need sreal
design.env.full <- read.csv("../data/CSI_Design_ENV_Source.csv", row.names=1)

# import taxonomy file for later
#csi.tax <- read.tax(taxonomy = "../data/CSI.0.03.cons.taxonomy", 
#                    format = "rdp", tax.levels = 6, col.tax = 3)
csi.tax <- read.tax(taxonomy = "../data/CSIvsearch.0.03.cons.taxonomy", 
                    format = "rdp", tax.levels = 6, col.tax = 3)
```

# manipulate data files for statistical analyses and graphing
```{r Bacterial Community Composition, include=FALSE}
# remove extra site (no mock community)

# design only with source tanks
missing <- setdiff(rownames(design.full), rownames(csi_otu))
design.full <- design.full[-(which(rownames(design.full) == missing)), ]
dim(design.full)

# OTU table - remove otu's w/ < 2 occurrences across all sites
otu_removal <- csi_otu[, which(colSums(csi_otu) >= 2)]
dim(otu_removal)

aa <- (rowSums(otu_removal))
aa # CSI033-7180 reads CSI101=75 reads - removed

# OTU table - removed low abundance samples
csi_low_remov <- otu_removal[which(rowSums(otu_removal) >= 13000), ]
dim(csi_low_remov)

# OTU table - odd sites in bacterial composition data and remove in design file
odd.sites <- c("CSI033","CSI101", "CSI130") #IUbarcode #CSI130 is mock community

otu_final <- csi_low_remov[setdiff(rownames(csi_low_remov), odd.sites), ]
design_final <- design.full[setdiff(rownames(design.full), odd.sites), ]
design.ns <- design.ns[setdiff(rownames(design.ns), odd.sites), ]

all.equal(rownames(design_final), rownames(otu_final))

# create presence/absence and relative abundance matrices
csi_pres_abs <- (otu_final > 0) * 1

csi_relabun <- otu_final
for (i in 1:dim(otu_final)[1]) {
  csi_relabun[i, ] <- otu_final[i, ]/sum(otu_final[i,])
}

# REMOVE source tanks from otu file
# REMOVE source tanks Number 1,2,3 from otu_final and use design.ns (manually removed source tanks)
temp <- rownames(design_final[which(design_final$Number %in% c(1,2,3)), ])
otu_final.ns <- otu_final[-(which(rownames(otu_final) %in% temp)), ]
dim(otu_final.ns)
dim(design.ns)

#drop missing data
missing <- setdiff(rownames(design.ns), rownames(otu_final.ns))
design.ns2 <- design.ns[-(which(rownames(design.ns) == missing)), ]
dim(design.ns2)

# Drop levels of factors that are no longer in data set 
design.ns.final <- droplevels(design.ns2)

all.equal(rownames(design.ns2), rownames(otu_final.ns))

# create presence/absence and relative abundance matrices
csi_pres_abs.ns <- (otu_final.ns > 0) * 1
csi_relabun.ns <- otu_final.ns
for (i in 1:dim(otu_final.ns)[1]) {
csi_relabun.ns[i, ] <- otu_final.ns[i, ]/sum(otu_final.ns[i,
])
}
# bind design and bact files NO Source
csi.full.ns <- cbind(design.ns.final,csi_relabun.ns)

csi.ns <- cbind(design.ns.final,otu_final.ns)

# bind design and bact files WITH SOURCE
csi.relabun.full <- cbind(design_final,csi_relabun)
dim(csi.relabun.full)

csi.PA.full <- cbind(design_final,csi_pres_abs)
dim(csi.PA.full)
# Set treatments
treatments1 <- as.factor(design.ns.final$Salinity)
levels(treatments1) <- c("0","5","9","13")
treatments2 <- as.factor(design.ns.final$Dispersal)
levels(treatments2) <- c("2","3")

#date_1 <- as.factor(desighn.ns.final$Date2)
```

# zooplankton community - bacterial community matrix comparisons
```{r zoop-bacteria comparisons}
#matrix comparison - Is there a relationship between zooplankton and bacterial community? = YES Mantel r = 0.409, P<0.0001
dim(otu_final.ns)

zoop.ns <- read.csv("../data/zoop_CSI_NoSource.csv", row.names=1)
#str(zoop.ns)
dim(zoop.ns)

#odd sites for bacteria CSI033 is ShortID ECU_CSI_034
#odd sites for bacteria CSI101 is Short ID	ECU_CSI_103
#deleted IUbarcode CSI041 = CSI 42 - double check
#CSI 27 has zero individuals after source tanks removed
odd.sites <- c("27","42","103") 

zoop.ns2 <- zoop.ns[setdiff(rownames(zoop.ns), odd.sites), ]
dim(zoop.ns2)

# Make Relative Abundance Matrices 
zoop.ns3 <- zoop.ns2[,-c(1:4)]
zoopREL <- zoop.ns3
for(i in 1:dim(zoop.ns3)[1]){
  zoopREL[i,] <- zoop.ns3[i,]/sum(zoop.ns3[i,])
}

# use csi_relabun.ns but drop CSI 27 sample/ IUbarcode ID CSI026	is CSI-027
#csi_relabun.ns
odd.sites <- c("CSI026")
dim(csi_relabun.ns)
csi_relabun.ns2 <- csi_relabun.ns[setdiff(rownames(csi_relabun.ns), odd.sites), ]

dim(csi_relabun.ns2)
dim(zoop.ns3)

dist.zoop <- vegdist(zoopREL, method = "bray")
dist.bact <- vegdist(csi_relabun.ns2, method = "bray")

require(ade4)
mantel.rtest(dist.zoop, dist.bact, nrepet = 999) #rsquared 0.411, p=.001
```

# Calculating Diversity Metrics
```{r (Diversity Metrics), echo=TRUE}
# Rarefy Abundances (min abundance is 13,240. We are sampling to 13,000) for uchime mothur run
# Rarefy Abundances (min abundance is 13,229. We are sampling to 13,000) for vsearch mothur run

min(rowSums(otu_final.ns))
otu.rarefy <- rrarefy(otu_final.ns, 13000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon <- diversity(otu.rarefy, "shannon")

# Species Richness
richness <- rowSums((otu.rarefy >= 1))

# Pielou’s evenness
J <- shannon/log(specnumber(otu.rarefy[,-c(1:1)]))

# Combined design,shannon,richness,evenness - no source tanks
csi_otu.div <- cbind(design.ns.final,shannon,richness,J)

# Rarefy Abundances (min abundance is 13,240. We are sampling to 13,000) for uchime mothur run
# Rarefy Abundances (min abundance is 13,229. We are sampling to 13,000) for vsearch mothur run
min(rowSums(otu_final))
otu.rarefy <- rrarefy(otu_final, 13000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon.source <- diversity(otu.rarefy, "shannon")

# Species Richness
richness.source <- rowSums((otu.rarefy >= 1))

# Pielou’s evenness
J.source <- shannon.source/log(specnumber(otu.rarefy[,-c(1:1)]))

# Combined design,shannon,richness,evenness - no source tanks
csi_otu.div.source <- cbind(design_final, shannon.source, 
                            richness.source,J.source)
```

# Testing salinity x dispersal influence on BACTERIAL DIVERSITY
```{r (Shannon), echo=TRUE}
# run full parametric statistical model
shannon.lm <- lm(shannon ~ Dispersal*Salinity*Date, data = csi_otu.div)
plot(shannon.lm)
shannon.lm
anova(shannon.lm)

# run linear regression measured salinity used instead of 'factor' salinity
shannon.reg <- lm(shannon~Salinity_real, data = csi_otu.div)
summary(shannon.reg)

# run for source tanks only
csi.div.source1 <- subset(csi_otu.div.source, Number == "1")
csi.div.source2 <- subset(csi_otu.div.source, Number == "2")
csi.div.source1 #salt source
csi.div.source2 #fresh source
csi.div.source <- rbind(csi.div.source1,csi.div.source2)
dim(csi.div.source)
Number1 <- as.factor(csi.div.source$Number)

shannon.lm.source <- lm(shannon.source ~ Date2, data = csi.div.source)
plot(shannon.lm.source)
summary(shannon.lm.source)
shannon.lm.source
anova(shannon.lm.source)

date_1 <- as.factor(design.ns.final$Date2)
labels <- c("0"="Day 0","18"="Day 18", "45"="Day 45")

# Graphing Shannon Diversity - Treatment Salinity NO SOURCE
p <- ggplot(csi_otu.div, aes(x=Salinity, y=shannon, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Date2)+facet_grid(. ~ Date2, labeller=labeller(Date2 = labels))
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Shannon Diversity Index (H')") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/ms/shannon.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

# Graphing Shannon Diversity - Treatment Salinity SOURCE ONLY
p <- ggplot(csi.div.source, aes(x=Date2, y=shannon.source, color=as.factor(Number1)))+ scale_color_manual(name="Dispersal Source", values=c("darkgray","blue"), labels = c("saltwater", "freshwater")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Day of Experiment", y = "Shannon Diversity Index (H')") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/ms/shannon.source.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

```

```{r (Evenness Metric), echo=TRUE}
# run full parametric statistical model
J.lm <- lm(J ~ Dispersal*Salinity*Date, data = csi_otu.div)
plot(J.lm)
library(car)
Anova(J.lm)

# run linear regression measured salinity used instead of 'factor' salinity
J.reg <- lm(J~Salinity_real, data = csi_otu.div)
summary(J.reg)

# run for source tanks only
J.lm.source <- lm(J.source ~ Number1, data = csi.div.source)
plot(J.lm.source)
summary(J.lm.source)
anova(J.lm.source)

# Graphing Pielous J - Treatment Salinity
p <- ggplot(csi_otu.div, aes(x=Salinity, y=J, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Date2)+facet_grid(. ~ Date2, labeller=labeller(Date2 = labels))
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Pielou's Evenness (J')") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/ms/pielousj.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

# Graphing Pielous J - Treatment Salinity SOURCE ONLY
p <- ggplot(csi.div.source, aes(x=Date2, y=J.source, color=as.factor(Number1)))+ scale_color_manual(name="Dispersal Source", values=c("darkgray","blue"), labels = c("saltwater", "freshwater")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Day of Experiment", y = "Pielou's Evenness (J')") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/ms/pielousj.source.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

```

```{r (Richness Metric), echo=TRUE}
# run full parametric statistical model
richness.lm <- lm(log(richness) ~ Dispersal*Salinity*Date, data = csi_otu.div)
plot(richness.lm)
richness.lm
anova(richness.lm)

# run linear regression measured salinity used instead of 'factor' salinity
richness.reg <- lm(richness~Salinity_real, data = csi_otu.div)
summary(richness.reg)

# run for source tanks only
richness.lm.source <- lm(richness.source ~ Number1, data = csi.div.source)
plot(richness.lm.source)
summary(richness.lm.source)
richness.lm.source
anova(richness.lm.source)

# Graphing richness - Treatment Salinity
p <- ggplot(csi_otu.div, aes(x=Salinity, y=richness, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Date2)+facet_grid(. ~ Date2, labeller=labeller(Date2 = labels))
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Species Richness") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/ms/richness.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

# Graphing richness - Treatment Salinity SOURCE ONLY
p <- ggplot(csi.div.source, aes(x=Date2, y=richness.source, color=as.factor(Number1)))+ scale_color_manual(name="Dispersal Source", values=c("darkgray","blue"), labels = c("saltwater", "freshwater")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Day of Experiment", y = "Species Richness") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/ms/richness.source.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```

```{r (Carbon Min), echo=FALSE}
#Cmin data inculded in full data set - no longer need to import separate file. Note: only use Date2=45 since Cmin only completed Day 45 of experiment

#subset according to Date2=45 only b/c Cmin only measured for Date2=45
csi.full.ns.Cmin  <- csi.full.ns[which(csi.full.ns$Date2=="45"),]
dim(csi.full.ns)
dim(csi.full.ns.Cmin)

# run full parametric statistical model
Cmin.lm <- lm(Cmin ~ Dispersal*Salinity, data = csi.full.ns.Cmin)
Cmin.lm
anova(Cmin.lm)

# run linear regression measured salinity used instead of 'factor' salinity
Cmin.reg <- lm(Cmin~Salinity_real, data = csi.full.ns.Cmin)
summary(Cmin.reg)

p <- ggplot(csi.full.ns.Cmin, aes(x=Salinity, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/ms/carbon_mineralization.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(csi.full.ns.Cmin, aes(x=Salinity_real, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/ms/real.sal.carbonmin.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

#graph Cmin/Shannon day 45 only
csi_otu.div.day45 <- subset(csi_otu.div, Date2 == "45")

p <- ggplot(csi_otu.div.day45, aes(x=shannon, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Shannon Diversity Index (H')", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/ms/Cmin_shannon.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

#graph Cmin/richness day 45 only

p <- ggplot(csi_otu.div.day45, aes(x=richness, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Species Richness", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/ms/Cmin_richness.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

```

```{r (Community Composition), echo=FALSE}
require("stats")
require("labdsv")
require("mgcv")
require("cluster")
require("vegan")
require("reshape2")
require("reshape")

#csi.full <- cbind(design_final,csi_relabun) - ready for stats
#csi.full.ns <- cbind(design.ns.final,csi_relabun.ns) - ready for stats NO SOURCE TANKS

sampleREL.dist1 <- vegdist(csi_relabun.ns, method="bray")

# PERMANOVA
adonis = adonis(csi.full.ns[,-c(1:16)] ~ Date2*Dispersal*Salinity, method = "bray", data = csi.full.ns, perm=1000)
adonis

# Principal Coordinates Analysis - NO SOURCE
CSI_pcoa1 <- cmdscale(sampleREL.dist1, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(CSI_pcoa1$eig[1] / sum(CSI_pcoa1$eig), 3) * 100
explainvar2a <- round(CSI_pcoa1$eig[2] / sum(CSI_pcoa1$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #17.3
explainvar2a #7.3
```
# PCoA Plot Design without Source
```{r (PCoA), include=FALSE}
all.equal(rownames(design.ns.final), rownames(csi_relabun.ns))

pcoa.groups <- paste(design.ns.final$Date, design.ns.final$Salinity, sep = "_")

pcoa.points <- data.frame(CSI_pcoa1$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Salinity
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Date
```

# Plot (ggplot)
```{r (Plot PCoA no source), echo=FALSE}
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                 group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col), colour = "black", size=6, stroke = 0.75) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  #scale_fill_manual(labels = c("0","5","9","13"), 
                    #values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  coord_cartesian(xlim = c(-0.25, 0.5), ylim = c(-0.35, 0.3)) + 
  theme(axis.title = element_text(size=18), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (17.3%)") + ylab("PCoA 2 (7.3%)") + 
  labs(fill = "Salinity", shape = "Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/ms/16SrRNA_CSI_Rplot_SalinityDispersal_NoSource.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```
# PCoA Plot Design (salinity x dispersal) without Source
```{r (PCoA), include=FALSE}
all.equal(rownames(design.ns.final), rownames(csi_relabun.ns))

pcoa.groups <- paste(design.ns.final$Dispersal, design.ns.final$Salinity, sep = "_")

pcoa.points <- data.frame(CSI_pcoa1$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Salinity
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Date
```

# Plot (ggplot) - incorporate date/dispersal/salinity for PCOA
```{r (Plot PCoA no source x dispersal), echo=FALSE}
all.equal(rownames(design.ns.final), rownames(csi_relabun.ns))

pcoa.groups <- paste(design.ns.final$Date, design.ns.final$Dispersal, design.ns.final$Salinity, sep = "_")

pcoa.points <- data.frame(CSI_pcoa1$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Salinity
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Dispersal

df1a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                 group = interaction(Date2, pcoa.col, pcoa.shape))) + theme_bw() 
plot1b <- plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col), colour = "black", size=6, stroke = 0.75) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  #scale_fill_manual(labels = c("0","5","9","13"), 
                    #values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_shape_manual(labels = c("fresh+salt (2)","salt only (3)"),
                     values = c(21, 24)) +
  coord_cartesian(xlim = c(-0.25, 0.5), ylim = c(-0.35, 0.3)) + 
  theme(axis.title = element_text(size=18), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (17.3%)") + ylab("PCoA 2 (7.3%)") + 
  labs(fill = "Salinity", shape = "Dispersal") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

plot1b + facet_wrap(~Date, ncol=1) +facet_grid(. ~ Date2, labeller=labeller(Date2 = labels)) #facet_wrap(~Date2, ncol = 1) 
plot1
ggsave("../figures/ms/16SrRNA_CSI_Rplot_SalinityDispersal_NoSource.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```

#PCoA sources only
```{r (Community Composition - source only), include=FALSE}
# run for source tanks only
csi.div.source1 <- subset(csi.relabun.full, Number == "1")
csi.div.source2 <- subset(csi.relabun.full, Number == "2")
csi.div.source1 #salt source
csi.div.source2 #fresh source
csi.relabun.full2 <- rbind(csi.div.source1,csi.div.source2)

sampleREL.dist2 <- vegdist(csi.relabun.full2[,-c(1:8)], method="bray")

# PERMANOVA
adonis2 <- adonis(csi.relabun.full2[,-c(1:8)] ~ Date*Dispersal*Salinity, method = "bray", data = csi.relabun.full2, perm=1000)
adonis2

# Principal Coordinates Analysis - WITH SOURCE
CSI_pcoa2 <- cmdscale(sampleREL.dist2, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(CSI_pcoa2$eig[1] / sum(CSI_pcoa2$eig), 3) * 100
explainvar2b <- round(CSI_pcoa2$eig[2] / sum(CSI_pcoa2$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #29.1
explainvar2b #9.8
```

#PCoA plot sources only
```{r (PCoA sources only), include=FALSE}
#all.equal(rownames(csi.relabun.full2), rownames(csi_relabun))

pcoa.groups2 <- paste(csi.relabun.full2$Date, csi.relabun.full2$Number, sep = "_")

pcoa.points2 <- data.frame(CSI_pcoa2$points, group = pcoa.groups2)

# Calculate Centroids (mean and SE)
pcoa.L.centroids2 <- melt(pcoa.points2, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids2 <- acast(pcoa.L.centroids2, variable ~ group, mean)
pcoa.centroids.se2 <- acast(pcoa.L.centroids2, variable ~ group, se)
pcoa.centroids.sd2 <- acast(pcoa.L.centroids2, variable ~ group, sd)

# Combine
pcoa.cent.dataframe2 <- cbind(t(pcoa.centroids2), t(pcoa.centroids.se2))
colnames(pcoa.cent.dataframe2) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats2 <- rownames(pcoa.cent.dataframe2)

pcoa.col2 <- as.factor(sapply(strsplit(pcoa.cent.treats2, "_"), `[`, 2)) # Tank Number
pcoa.shape2 <- as.factor(sapply(strsplit(pcoa.cent.treats2, "_"), `[`, 1))  # Date
```

# PLOT GGPLOT sources only
```{r (PCoA plot sources only), echo=FALSE}
df1a2 <- as.data.frame(pcoa.cent.dataframe2)
plot1a2 <- ggplot(df1a2, aes(x=V1, y=V2, colour=pcoa.col2, shape = pcoa.shape2,
                 group = interaction(pcoa.col2, pcoa.shape2))) + theme_bw() 
plot1a2 + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col2), colour = "black", size=6, stroke = 0.75) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(labels = c("saltwater","freshwater"), 
                      values = c("#FFFFFF", "#0066CC")) + 
 scale_fill_manual(labels = c("saltwater","freshwater"), 
                     values = c("#FFFFFF", "#0066CC")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  theme(axis.title = element_text(size=18), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (29.1%)") + ylab("PCoA 2 (9.8%)") + 
  labs(fill = "Dispersal Source", shape = "Day of Experiment") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/ms/16SrRNA_CSI_Rplot_Source.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```

```{r (structure-function without source), echo=TRUE}
#Is there a relationship betwen bacterial community composition and decomposition rate? used matrix comparison Mantel test
#str(csi.full.ns.Cmin)
df.bcc <- csi.full.ns.Cmin[,-c(1:16)] #bacteria
df.env.bcc <- csi.full.ns.Cmin[,c(1:16)] #env
df.decomp <- csi.full.ns.Cmin[,c(12:14)] #decomp
df.decompCmin <- csi.full.ns.Cmin[,c(12:15)] #decompCmin

#bray curtis for bact comm
dist.bcc <- vegdist(df.bcc[,-1], method = "bray")
#euclidean distance for leaf litter decomp
dist.decomp <- vegdist(df.decomp, method = "euclidean")
#euclidean distance for leaf litter decomp and Cmin
dist.decompCmin <- vegdist(df.decompCmin, method = "euclidean")
#euclidean distance for cmin
cmin.df <- df.env.bcc %>% dplyr::select(Cmin)
dist.cmin <- vegdist(cmin.df,method = "euclidean")

#matrix comparison cmin and microbial composition
mantel.rtest(dist.bcc, dist.cmin, nrepet = 999) # r=0.22, p=0.006

#matrix comparison
mantel.rtest(dist.bcc, dist.decomp, nrepet = 999) #bacteria compared to decomp correlation r = 0.231, P=0.006

mantel.rtest(dist.bcc, dist.decompCmin, nrepet = 999) #bacteria compared to decompCmin correlation r = 0.212, P=0.007

#Redundancy Analysis
f <- df.bcc ~ Maple_dmass + Spartina_dmass + Phrag_dmass + Cmin
df.rda <- dbrda(f, data=df.env.bcc, distance="bray")
anova(df.rda, by="terms", model="direct")
RsquareAdj(df.rda)

plot(df.rda)
```
# Indicator Species Analysis
```{r}
library("labdsv")
#http://rfunctions.blogspot.com/2013/02/multivariate-analysis-indicator-value.html

# bind design and bact files NO Source
csi.full.ns <- cbind(design.ns.final,csi_relabun.ns)

csi.ns <- cbind(design.ns.final,otu_final.ns)

# Set treatments
treatments1 <- as.factor(csi.full.ns$Salinity)
levels(treatments1) <- c("0","5","9","13")

indval.default <- function(x,clustering,numitr=1000, ...)
{
    if (!is.data.frame(x)) x <- data.frame(x)

    clustering <- clustify(clustering)
    levels <- levels(clustering)
    clustering <- as.integer(clustering)

    if (any(apply(x>0,2,sum)==0)) stop('All species must occur in at least one plot')
    numplt <- nrow(x)
    numspc <- ncol(x)
    numcls <- as.integer(length(table(clustering)))
    maxcls <- rep(0,numspc)
    relfrq <- matrix(0,nrow=numspc,ncol=numcls)
    relabu <- matrix(0,nrow=numspc,ncol=numcls)
    indval <- matrix(0,nrow=numspc,ncol=numcls)
    indcls <- rep(0,numspc)
    pval <- rep(0,numspc)
    tmpfrq <- rep(0.0,numcls)
    tmpabu <- rep(0.0,numcls)
    pclass <- rep(0,numplt)
    tclass <- rep(0,numplt)
    errcod <- 0
    tmp <- .Fortran("duleg",
        as.double(as.matrix(x)),
        as.integer(numplt),
        as.integer(numspc),
        as.integer(factor(clustering)),
        as.integer(table(clustering)),
        as.integer(numcls),
        as.integer(numitr),
        relfrq = relfrq,
        relabu = relabu,
        indval = indval,
        pval = pval,
        indcls = indcls,
        maxcls = as.integer(maxcls),
        as.double(tmpfrq),
        as.double(tmpabu),
        as.integer(pclass),
        as.integer(tclass),
        errcod = as.integer(errcod),
        PACKAGE='labdsv')
    out <- list(relfrq=data.frame(tmp$relfrq),relabu=data.frame(tmp$relabu),
              indval=data.frame(tmp$indval),maxcls=tmp$maxcls,indcls=tmp$indcls,
              pval=tmp$pval,error=tmp$errcod)
    row.names(out$relfrq) <- names(x)
    row.names(out$relabu) <- names(x)
    row.names(out$indval) <- names(x)
    names(out$maxcls) <- names(x)
    names(out$indcls) <- names(x)
    names(out$pval) <- names(x)
    names(out$relfrq) <- levels
    names(out$relabu) <- levels
    names(out$indval) <- levels
    class(out) <- 'indval'
    if (out$error == 1) cat('WARNING: one or more species not assigned to any cluster\n')
    out
}

csi_relabun.ns <- csi_relabun.ns[, colSums(csi_relabun.ns) > 0.05]
bac.ind <- indval(csi_relabun.ns, treatments1)
levels(treatments1)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- csi.tax[which(as.character(csi.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(treatments1)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators.txt",
            sep="\t", row.names = F, quote = F)

split(indicator.bac$Phylum, indicator.bac$Cluster)
split(indicator.bac$Class, indicator.bac$Cluster)

plot(as.factor(indicator.bac$Cluster), as.factor(indicator.bac$Phylum))

```

# Phylogenetic Analysis
```{r}
library("ggtree"); library("ape")
tree <- read.tree("../data/CSI.bac.rename.tree")

# otu_final, design_final
design.45 <- design_final[which(design_final$Date2 == "45"), ]
design.45 <- design.45[which(design.45$Dispersal %in% c(2, 3)), ]
otus.45 <- otu_final[which(rownames(otu_final) %in% rownames(design.45)), ]
otus.45 <- otus.45[, which(colSums(otus.45) > 100)]

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in% 
                                csi.tax$OTU[csi.tax$Phylum %in%
                                c("Bacteria_unclassified", "OD1", 
                                  "BRC1", "Spirochaetes", 
                                  "Tenericutes", "Deferribacteres",
                                  "Chloroflexi", "Deinococcus-Thermus", "WS3")]))]

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in% 
                                csi.tax$OTU[csi.tax$Class %in%
                                c("Bacteroidetes_unclassified",
                           "Bacteroidetes_incertae_sedis_class_incertae_sedis",
                           "Proteobacteria_unclassified", "WS3", "Bacteroidia", "Chlorobi", "Epsilonproteobacteria")]))]

tree$tip.label <- gsub(".{1}$", "", tree$tip.label)

tree.labs <- tree$tip.label
tree.labs %in% colnames(otus.45)

rm.otus <- setdiff(tree.labs, colnames(otus.45))

tree.2 <- drop.tip(tree, rm.otus)

str(tree.2)

length(tree.2$tip.label)
sum(tree.2$tip.label %in% colnames(otus.45))

tree.otus <- tree.2$tip.label
tree.tax <- csi.tax[which(csi.tax$OTU %in% tree.otus), ]
table(tree.tax$Class)

tree.tax.2 <- tree.tax[order(match(tree.tax$OTU, tree.2$tip.label)), ]
all.equal(as.character(tree.tax.2$OTU), tree.2$tip.label)

phylum2 <- tree.tax.2$Phylum
for(i in 1:length(phylum2)){
  if (phylum2[i] == "Proteobacteria"){
    phylum2[i] <- tree.tax.2$Class[i]
  }
  if (phylum2[i] == "Bacteroidetes"){
    phylum2[i] <- tree.tax.2$Class[i]
  }
}

table(phylum2)

groupInfo <- split(tree.2$tip.label, phylum2)
Phy.tree <- groupOTU(tree.2, groupInfo, group_name = "phylum2")
levels(attributes(Phy.tree)$phylum2) <- c(names(groupInfo))

p <- ggtree(Phy.tree, layout="rectangular") + theme(legend.position="top", 
        legend.key = element_rect(colour = NA)) + 
  geom_tippoint(aes(color=phylum2)) 


# Set treatments
treatments1 <- as.factor(design.45$Salinity) 
levels(treatments1) <- c("0","5","9","13")

sal.ind <- indval(otus.45, treatments1)
levels(treatments1)
summary(sal.ind)
sal.inds <- which(sal.ind$pval <= 0.05)
sal.indicators <- as.data.frame(matrix(NA, nrow = length(sal.inds), ncol = 4))
colnames(sal.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
sal.indicators$OTU <- names(sal.inds)
sal.indicators$Cluster <- sal.ind$maxcls[sal.inds]
sal.indicators$IndVal <- sal.ind$indcls[sal.inds]
sal.indicators$Prob <- sal.ind$pval[sal.inds]

sal.dat <- as.data.frame(matrix(NA, ncol = 5, nrow = length(tree.2$tip.label)))
rownames(sal.dat) <- tree.2$tip.label
colnames(sal.dat) <- c(levels(treatments1), ">0")

for(i in 1:dim(sal.indicators)[1]){
  temp <- sal.indicators[i, ]
  sal.dat[which(row.names(sal.dat) == temp$OTU), temp$Cluster] <- 10
}


treatments1 <- as.factor(design.45$Salinity) 
levels(treatments1) <- c("0","1","1","1")
sal.ind <- indval(otus.45, treatments1)
levels(treatments1)
summary(sal.ind)
sal.inds <- which(sal.ind$pval <= 0.05)
sal.indicators <- as.data.frame(matrix(NA, nrow = length(sal.inds), ncol = 4))
colnames(sal.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
sal.indicators$OTU <- names(sal.inds)
sal.indicators$Cluster <- sal.ind$maxcls[sal.inds]
sal.indicators$IndVal <- sal.ind$indcls[sal.inds]
sal.indicators$Prob <- sal.ind$pval[sal.inds]

for(i in 1:dim(sal.indicators)[1]){
  temp <- sal.indicators[i, ]
  if(temp$Cluster == 2){
    sal.dat[which(row.names(sal.dat) == temp$OTU), 5] <- 10
  }
}


treatments2 <- droplevels(as.factor(design.45$Dispersal))
levels(treatments2) <- c("2", "3")

dis.ind <- indval(otus.45, treatments2)
levels(treatments2)
summary(dis.ind)
dis.inds <- which(dis.ind$pval <= 0.05)
dis.indicators <- as.data.frame(matrix(NA, nrow = length(dis.inds), ncol = 4))
colnames(dis.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
dis.indicators$OTU <- names(dis.inds)
dis.indicators$Cluster <- dis.ind$maxcls[dis.inds]
dis.indicators$IndVal <- dis.ind$indcls[dis.inds]
dis.indicators$Prob <- dis.ind$pval[dis.inds]

csi.tax[which(csi.tax$OTU %in% dis.indicators$OTU), ]
csi.tax[which(csi.tax$OTU %in% sal.indicators$OTU), ]

gheatmap(p, sal.dat, width = 0.5, colnames_angle=90, hjust=1, 
         low = "red", high = "red")

ggsave("../figures/indicator.tree.pdf", width = 10, height = 8, units = "in")

```








