---
title: "Salinity affects bacterial phylogeny-function"
author: "Alexandra Stucy, Mario Muscarella, Jo Werba, Mike McCoy, Ariane Peralta"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document: null
  fig_caption: yes
  html_document: default
editor_options:
  chunk_output_type: console
---
Project Description: Analysis of salinity and dispersal of saline aquatic communities on bacterial community structure and function. Manuscript preparation for zooplankton/bacterioplankton community composition-function study.

# Setting up working directory, packages
```{r Setup, include=FALSE}
setwd("~/GitHub/CSI_Dispersal/analyses/")
rm(list = ls())

# Set Source R Tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

# Load Req'd Packages
library("vegan"); library("ecodist"); library("Hmisc")
library("dplyr"); library("reshape2"); library("labdsv")
library("nlme"); library("MASS"); library("ade4")
library("png"); library("grid"); library("ggplot2")
library("ape"); library("picante"); library("ggtree")

# Set Std Err Functions
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

ci <- function(x, ...) {
1.96 * sd(x, na.rm = TRUE)
}
```

# Loading data file
```{r Loading Files, include=FALSE}
# NOTES: Tank 3 only had fresh water added 
#        Tank 1 was the source for the salt  
#        Tank 2 was the source for the fresh water
#
# NOTES: Dispersal 0 (treatments 1 and 2) were source tanks-- no dispersal.
#        Dispersal 1 (treatment 3) only received fresh water 
#        Dispersal 3 (treatments 4,6,8,10) only got salt water
#        Dispersal 2 (treatments 5,7,9,11) received both fresh and salt

# Load design file - no source tanks
design.ns <- read.csv("../data/CSI_Design_ENV_NoSourceTanks.csv", row.names=1)
#head(design.ns)
#str(design.ns)
dim(design.ns)

# Load design file - with source tanks but 
#                    missing decomp, nutrients, sreal, Date2 columns
design.full <- read.csv("../data/design_CSI.csv", row.names=1)
design.full <- design.full[-c(grep("mock community", design.full$CSI_ID)), ]
dim(design.full)

# load OTU file
csi_otu <- read.otu("../mothur/CSI.bac.final.shared")
dim(csi_otu)

# load environ/function file - 
# includes expt design, decomp data, Cmin, nutrients day 45 only - need sreal
design.env.full <- read.csv("../data/CSI_Design_ENV_Source.csv", row.names=1)

# Import Taxonomy File for later
csi.tax <- read.tax(taxonomy = "../mothur/CSI.bac.final.taxonomy", 
                    format = "rdp", tax.levels = 6, col.tax = 3)
csi.tax$Phylum[which(csi.tax$Phylum == "Cyanobacteria/Chloroplast")] <- "Cyanobacteria"

# Import Tree File
tree <- read.tree("../mothur/CSI.bac.rename.tree")

# Import UniFrac Distnce Matrix
UniFrac.raw <- read.delim("../mothur/CSI.bac.tree1.weighted.phylip.dist", skip = 1, header = F, row.names = 1)
row.names(UniFrac.raw) <- gsub("    ", "", row.names(UniFrac.raw))
colnames(UniFrac.raw) <- rownames(UniFrac.raw)
```

# Manipulate data files for statistical analyses and graphing
```{r Bacterial Community Composition, include=FALSE}
# remove extra site (no mock community)

# design only with source tanks
missing <- setdiff(rownames(design.full), rownames(csi_otu))
design.full <- design.full[-(which(rownames(design.full) == missing)), ]
dim(design.full)

# OTU table - remove otu's w/ < 2 occurrences across all sites
otu_removal <- csi_otu[, which(colSums(csi_otu) >= 2)]
dim(otu_removal)

aa <- (rowSums(otu_removal))
aa # CSI033-7180 reads CSI101=75 reads - removed

# OTU table - removed low abundance samples
csi_low_remov <- otu_removal[which(rowSums(otu_removal) >= 13000), ]
dim(csi_low_remov)

# OTU table - odd sites in bacterial composition data and remove in design file
odd.sites <- c("CSI033","CSI101", "CSI130") #IUbarcode #CSI130 is mock community

otu_final <- csi_low_remov[setdiff(rownames(csi_low_remov), odd.sites), ]
design_final <- design.full[setdiff(rownames(design.full), odd.sites), ]
design.ns <- design.ns[setdiff(rownames(design.ns), odd.sites), ]

all.equal(rownames(design_final), rownames(otu_final))

# create presence/absence and relative abundance matrices
csi_pres_abs <- (otu_final > 0) * 1

csi_relabun <- otu_final
for (i in 1:dim(otu_final)[1]) {
  csi_relabun[i, ] <- otu_final[i, ]/sum(otu_final[i,])
}

# REMOVE source tanks from otu file
# REMOVE source tanks Number 1,2,3 from otu_final and use design.ns (manually removed source tanks)
temp <- rownames(design_final[which(design_final$Number %in% c(1,2,3)), ])
otu_final.ns <- otu_final[-(which(rownames(otu_final) %in% temp)), ]
otu_final.ns <- otu_final.ns[, which(colSums(otu_final.ns) > 0)]
dim(otu_final.ns)
dim(design.ns)

#drop missing data
missing <- setdiff(rownames(design.ns), rownames(otu_final.ns))
design.ns2 <- design.ns[-(which(rownames(design.ns) == missing)), ]
dim(design.ns2)

# Drop levels of factors that are no longer in data set 
design.ns.final <- droplevels(design.ns2)

all.equal(rownames(design.ns2), rownames(otu_final.ns))

# create presence/absence and relative abundance matrices
csi_pres_abs.ns <- (otu_final.ns > 0) * 1
csi_relabun.ns <- otu_final.ns
for (i in 1:dim(otu_final.ns)[1]) {
  csi_relabun.ns[i, ] <- otu_final.ns[i, ]/sum(otu_final.ns[i, ])
}
# bind design and bact files NO Source
csi.full.ns <- cbind(design.ns.final,csi_relabun.ns)

csi.ns <- cbind(design.ns.final,otu_final.ns)

# bind design and bact files WITH SOURCE
csi.relabun.full <- cbind(design_final,csi_relabun)
dim(csi.relabun.full)

csi.PA.full <- cbind(design_final,csi_pres_abs)
dim(csi.PA.full)
# Set treatments
treatments1 <- as.factor(design.ns.final$Salinity)
levels(treatments1) <- c("0","5","9","13")
treatments2 <- as.factor(design.ns.final$Dispersal)
levels(treatments2) <- c("2","3")

#date_1 <- as.factor(desighn.ns.final$Date2)
```

# Calculating OTU Diversity Metrics [included in Werba et al. (In review) manuscript]
```{r (Diversity Metrics), echo=TRUE}
# Rarefy Abundances (min abundance is 13,240. We are sampling to 13,000) for uchime mothur run
# Rarefy Abundances (min abundance is 13,229. We are sampling to 13,000) for vsearch mothur run

min(rowSums(otu_final.ns))
otu.rarefy <- rrarefy(otu_final.ns, 20000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon <- diversity(otu.rarefy, "shannon")

# Species Richness
richness <- rowSums((otu.rarefy >= 1))

# Pielou’s evenness
J <- shannon/log(specnumber(otu.rarefy[,-c(1:1)]))

# Combined design,shannon,richness,evenness - no source tanks
csi_otu.div <- cbind(design.ns.final,shannon,richness,J)

# Rarefy Abundances (min abundance is 13,240. We are sampling to 13,000) for uchime mothur run
# Rarefy Abundances (min abundance is 13,229. We are sampling to 13,000) for vsearch mothur run
min(rowSums(otu_final))
otu.rarefy <- rrarefy(otu_final, 17000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon.source <- diversity(otu.rarefy, "shannon")

# Species Richness
richness.source <- rowSums((otu.rarefy >= 1))

# Pielou’s evenness
J.source <- shannon.source/log(specnumber(otu.rarefy[,-c(1:1)]))

# Combined design,shannon,richness,evenness - no source tanks
csi_otu.div.source <- cbind(design_final, shannon.source, 
                            richness.source,J.source)
```

# Graph Cmin/Shannon day 45 only
```{r (Cmin/Shannon day 45 only), echo=FALSE}
csi_otu.div.day45 <- subset(csi_otu.div, Date2 == "45")

p <- ggplot(csi_otu.div.day45, aes(x=shannon, y=Cmin, 
                                   color = as.factor(Dispersal))) +
  scale_color_manual(name="Dispersal Treatment", 
                     values=c("black","gray"), 
                     labels = c("fresh+salt (2)", "salt only (3)")) # +
  # stat_summary(fun.data = "mean_cl_boot", size=1) 
p1 = p + geom_smooth(method="lm", se=FALSE) + geom_point(size=4) 
p1 + theme_bw() + theme(panel.grid.major = element_blank(), 
                        panel.grid.minor = element_blank(), 
                        axis.line = element_line(colour = "black")) + 
  theme(axis.title=element_text(vjust=1,size=16,face="bold"), 
        axis.text = element_text(size=16), 
        axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), 
        panel.border = element_rect(colour = "black",size=1)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  labs(x = "Shannon Diversity Index (H')", 
       y = (expression(paste("C Mineralization (ng CO"[2],
                             " mL"^-{1},"day"^-{1},")")))) + 
  theme(strip.text.x = element_text(size=16, face="bold"), 
        strip.text.y = element_text(size=16, face="bold"), 
        strip.background = element_rect(colour="black", 
                                        fill="white", size=1))

ggsave("../figures/Cmin_shannon.pdf", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

#graph Cmin/richness day 45 only

p <- ggplot(csi_otu.div.day45, 
            aes(x=richness, y=Cmin, color=as.factor(Dispersal))) + 
  scale_color_manual(name="Dispersal Treatment", 
                     values=c("black","gray"), 
                     labels = c("fresh+salt (2)", "salt only (3)")) 
# +  stat_summary(fun.data="mean_cl_boot",size=1) 
p1 = p + geom_smooth(method="lm", se=FALSE) + geom_point(size=4)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), 
                        panel.grid.minor = element_blank(), 
                        axis.line = element_line(colour = "black")) + 
  theme(axis.title = element_text(vjust=1,size=16,face="bold"), 
        axis.text = element_text(size=16), 
        axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), 
        panel.border = element_rect(colour = "black",size=1)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  labs(x = "Species Richness", 
       y = (expression(paste("C Mineralization (ng CO"[2],
                             " mL"^-{1}," day"^-{1},")")))) + 
  theme(strip.text.x = element_text(size=16, face="bold"), 
        strip.text.y = element_text(size=16, face="bold"), 
        strip.background = element_rect(colour="black", 
                                        fill="white", size=1))

ggsave("../figures/Cmin_richness.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

```

# Phylogenetic Indicator Figure Analysis
```{r}
str(tree)

# tree <- midpoint.root(tree)

# otu_final, design_final
design.45 <- design_final[which(design_final$Date2 == "45"), ]
design.45 <- design.45[which(design.45$Dispersal %in% c(2, 3)), ]
otus.45 <- otu_final[which(rownames(otu_final) %in% rownames(design.45)), ]
otus.45 <- otus.45[, which(colSums(otus.45) > 100)]

low.phy <- names(which(table(csi.tax$Phylum) < 20))
low.cla <- names(which(table(csi.tax$Class) < 20))

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in%
                                csi.tax$OTU[csi.tax$Phylum %in%
                                c(low.phy)]))]

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in%
                                csi.tax$OTU[csi.tax$Class %in%
                                c(low.cla)]))]


otus.45 <- otus.45[, -c(which(colnames(otus.45) %in% 
                                csi.tax$OTU[csi.tax$Phylum %in%
                                c("Acidobacteria", "Bacteria_unclassified",
                                  "OD1", "Parcubacteria", "Ignavibacteriae",
                                  "BRC1", "Spirochaetes", "Armatimonadetes", 
                                  "Tenericutes", "Deferribacteres",
                                  "Chlamydiae", "Gemmatimonadetes", 
                                  "WS3", "Firmicutes",
                                  "Chloroflexi", "Deinococcus-Thermus")]))]
otus.45 <- otus.45[, -c(which(colnames(otus.45) %in% 
                                csi.tax$OTU[csi.tax$Class %in%
                                c("Bacteroidetes_unclassified",
                           "Bacteroidetes_incertae_sedis_class_incertae_sedis",
                           "Proteobacteria_unclassified", "WS3", "Bacteroidia", "Chlorobi", "Epsilonproteobacteria")]))]

# Rename Error (Fixed in the actual file to prevent errors)
# tree$tip.label <- gsub(".{1}$", "", tree$tip.label)

tree.labs <- tree$tip.label
tree.labs %in% colnames(otus.45)

rm.otus <- setdiff(tree.labs, colnames(otus.45))

tree.2 <- drop.tip(tree, rm.otus)

str(tree.2)

length(tree.2$tip.label)
sum(tree.2$tip.label %in% colnames(otus.45))

tree.otus <- tree.2$tip.label
tree.tax <- csi.tax[which(csi.tax$OTU %in% tree.otus), ]
table(tree.tax$Phylum)

tree.tax.2 <- tree.tax[order(match(tree.tax$OTU, tree.2$tip.label)), ]
all.equal(as.character(tree.tax.2$OTU), tree.2$tip.label)

phylum2 <- tree.tax.2$Phylum
for(i in 1:length(phylum2)){
  if (phylum2[i] == "Proteobacteria"){
    phylum2[i] <- tree.tax.2$Class[i]
  }
}

#   if (phylum2[i] == "Bacteroidetes"){
#     phylum2[i] <- tree.tax.2$Class[i]
#   }
# }

table(phylum2)

groupInfo <- split(tree.2$tip.label, phylum2)
Phy.tree <- groupOTU(tree.2, groupInfo, group_name = "phylum2")
levels(attributes(Phy.tree)$phylum2) <- c("Root", names(groupInfo))

p <- ggtree(Phy.tree, layout="rectangular") + theme(legend.position="top", 
        legend.key = element_rect(colour = NA)) + 
  geom_tippoint(aes(color=phylum2)) 


# Set treatments
treatments1 <- as.factor(design.45$Salinity) 
levels(treatments1) <- c("0","5","9","13")

sal.ind <- indval(otus.45, treatments1)
levels(treatments1)
summary(sal.ind)
sal.inds <- which(sal.ind$pval <= 0.05)
sal.indicators <- as.data.frame(matrix(NA, nrow = length(sal.inds), ncol = 4))
colnames(sal.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
sal.indicators$OTU <- names(sal.inds)
sal.indicators$Cluster <- sal.ind$maxcls[sal.inds]
sal.indicators$IndVal <- sal.ind$indcls[sal.inds]
sal.indicators$Prob <- sal.ind$pval[sal.inds]

sal.dat <- as.data.frame(matrix(NA, ncol = 5, nrow = length(tree.2$tip.label)))
rownames(sal.dat) <- tree.2$tip.label
colnames(sal.dat) <- c(levels(treatments1), ">0")

for(i in 1:dim(sal.indicators)[1]){
  temp <- sal.indicators[i, ]
  sal.dat[which(row.names(sal.dat) == temp$OTU), temp$Cluster] <- 10
}


treatments1 <- as.factor(design.45$Salinity) 
levels(treatments1) <- c("0","1","1","1")
sal.ind <- indval(otus.45, treatments1)
levels(treatments1)
summary(sal.ind)
sal.inds <- which(sal.ind$pval <= 0.05)
sal.indicators <- as.data.frame(matrix(NA, nrow = length(sal.inds), ncol = 4))
colnames(sal.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
sal.indicators$OTU <- names(sal.inds)
sal.indicators$Cluster <- sal.ind$maxcls[sal.inds]
sal.indicators$IndVal <- sal.ind$indcls[sal.inds]
sal.indicators$Prob <- sal.ind$pval[sal.inds]

for(i in 1:dim(sal.indicators)[1]){
  temp <- sal.indicators[i, ]
  if(temp$Cluster == 2){
    sal.dat[which(row.names(sal.dat) == temp$OTU), 5] <- 10
  }
}


treatments2 <- droplevels(as.factor(design.45$Dispersal))
levels(treatments2) <- c("2", "3")

dis.ind <- indval(otus.45, treatments2)
levels(treatments2)
summary(dis.ind)
dis.inds <- which(dis.ind$pval <= 0.05)
dis.indicators <- as.data.frame(matrix(NA, nrow = length(dis.inds), ncol = 4))
colnames(dis.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
dis.indicators$OTU <- names(dis.inds)
dis.indicators$Cluster <- dis.ind$maxcls[dis.inds]
dis.indicators$IndVal <- dis.ind$indcls[dis.inds]
dis.indicators$Prob <- dis.ind$pval[dis.inds]

csi.tax[which(csi.tax$OTU %in% dis.indicators$OTU), ]
csi.tax[which(csi.tax$OTU %in% sal.indicators$OTU), ]

gheatmap(p, sal.dat, width = 0.5, colnames_angle=90, hjust=1, 
         low = "black", high = "black")+ theme(legend.position = "right") + guides(fill = FALSE)   

ggsave("../figures/indicator.tree.pdf", width = 10, height = 8, units = "in")

```

# Phylogenetic Diversity 
```{r (Phylogenetic Diversity), include=FALSE}
str(tree)
dim(otu_final)
dim(csi_otu)

sum(tree$tip.label %in% colnames(csi_otu) == FALSE) # Tree Represents All

# Small Branches
sum(tree$edge.length < 0.0000001)

# Import Unifrac Matrix
str(UniFrac.raw)
dim(UniFrac.raw)

UniFrac <- UniFrac.raw[which(row.names(UniFrac.raw) %in%
                                   row.names(otu_final)),
                             which(colnames(UniFrac.raw) %in%
                                   row.names(otu_final))]

dim(UniFrac)

# Make into Distance Matrix
unifrac.dist <- as.dist(UniFrac, upper = T, diag = T)

# Calculate Phylo Diversity
phylo.d <- pd(otu_final, tree, include.root = F)
phylo_final <- phylo.d[setdiff(rownames(phylo.d), odd.sites), ]
#design_final <- design.full[setdiff(rownames(design.full), odd.sites), ]
#design.ns <- design.ns[setdiff(rownames(design.ns), odd.sites), ]
all.equal(rownames(design_final), rownames(phylo_final))

# REMOVE source tanks from otu file
# REMOVE source tanks Number 1,2,3 from otu_final and use design.ns (manually removed source tanks)
temp <- rownames(design_final[which(design_final$Number %in% c(1,2,3)), ])
phylo_final.ns <- phylo_final[-(which(rownames(phylo_final) %in% temp)), ]
dim(phylo_final.ns)
dim(design.ns.final)

csi.phylo.ns <- cbind(design.ns.final,phylo_final.ns)
```

# Phylogenetic Linear Models
```{r (Phylogenetic Diversity - plot), echo=TRUE}
# run full parametric statistical model
PD.lm <- lm(PD ~ Dispersal*Salinity*Date, data = csi.phylo.ns)
plot(PD.lm)
anova(PD.lm)

# run linear regression measured salinity used instead of 'factor' salinity
PD.reg <- lm(PD~Salinity_real, data = csi.phylo.ns)
summary(PD.reg)

# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+facet_wrap(~Date2)+facet_grid(. ~ Date2)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/phylogenetic_div.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

csi.phylo.ns.day45 <- subset(csi.phylo.ns, Date2 == "45")

PD.lm.45 <- lm(PD ~ Dispersal*Salinity, data = csi.phylo.ns.day45)
anova(PD.lm.45)

# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns.day45, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/phylogenetic_div.45.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(csi.phylo.ns.day45, aes(x=PD, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+geom_point(size=4)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Phylogenetic Diversity", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/Cmin_PD.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```


# Ordination - based on unifrac
```{r}
# Principal Coordinates Analysis - NO SOURCE
CSI_pcoa_phylo <- cmdscale(unifrac.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1 <- round(CSI_pcoa_phylo$eig[1] / sum(CSI_pcoa_phylo$eig), 3) * 100
explainvar2 <- round(CSI_pcoa_phylo$eig[2] / sum(CSI_pcoa_phylo$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2)

explainvar1 
explainvar2


all.equal(rownames(design_final), labels(unifrac.dist))

pcoa.groups <- paste(design_final$Date, design_final$Salinity, sep = "_")

pcoa.points <- data.frame(CSI_pcoa_phylo$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Salinity
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Date

df1a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                 group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col), colour = "black", size=6, stroke = 1) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  #scale_fill_manual(labels = c("0","5","9","13"), 
                    #values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  coord_cartesian(xlim = c(-0.25, 0.25), ylim = c(-0.2, 0.2)) + 
  theme(axis.title = element_text(size=20), axis.text=element_text(size=18), 
          axis.text.x = element_text(size=18), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (17.3%)") + ylab("PCoA 2 (7.3%)") + 
  labs(fill = "Salinity", shape = "Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 5, colour="black")),
         shape = guide_legend(override.aes = list(size = 5, fill="black")))

ggsave("../figures/Phylo_Ordination.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```

# Phylogenetic Regression



```
# Indicator Species Analysis [included in Werba et al. (In review) manuscript]
```{r}
library("labdsv")
#http://rfunctions.blogspot.com/2013/02/multivariate-analysis-indicator-value.html

# bind design and bact files NO Source
csi.full.ns <- cbind(design.ns.final,csi_relabun.ns)

csi.ns <- cbind(design.ns.final,otu_final.ns)

# Set treatments
treatments1 <- as.factor(csi.full.ns$Salinity)
levels(treatments1) <- c("0","5","9","13")

indval.default <- function(x,clustering,numitr=1000, ...)
{
    if (!is.data.frame(x)) x <- data.frame(x)

    clustering <- clustify(clustering)
    levels <- levels(clustering)
    clustering <- as.integer(clustering)

    if (any(apply(x>0,2,sum)==0)) stop('All species must occur in at least one plot')
    numplt <- nrow(x)
    numspc <- ncol(x)
    numcls <- as.integer(length(table(clustering)))
    maxcls <- rep(0,numspc)
    relfrq <- matrix(0,nrow=numspc,ncol=numcls)
    relabu <- matrix(0,nrow=numspc,ncol=numcls)
    indval <- matrix(0,nrow=numspc,ncol=numcls)
    indcls <- rep(0,numspc)
    pval <- rep(0,numspc)
    tmpfrq <- rep(0.0,numcls)
    tmpabu <- rep(0.0,numcls)
    pclass <- rep(0,numplt)
    tclass <- rep(0,numplt)
    errcod <- 0
    tmp <- .Fortran("duleg",
        as.double(as.matrix(x)),
        as.integer(numplt),
        as.integer(numspc),
        as.integer(factor(clustering)),
        as.integer(table(clustering)),
        as.integer(numcls),
        as.integer(numitr),
        relfrq = relfrq,
        relabu = relabu,
        indval = indval,
        pval = pval,
        indcls = indcls,
        maxcls = as.integer(maxcls),
        as.double(tmpfrq),
        as.double(tmpabu),
        as.integer(pclass),
        as.integer(tclass),
        errcod = as.integer(errcod),
        PACKAGE='labdsv')
    out <- list(relfrq=data.frame(tmp$relfrq),relabu=data.frame(tmp$relabu),
              indval=data.frame(tmp$indval),maxcls=tmp$maxcls,indcls=tmp$indcls,
              pval=tmp$pval,error=tmp$errcod)
    row.names(out$relfrq) <- names(x)
    row.names(out$relabu) <- names(x)
    row.names(out$indval) <- names(x)
    names(out$maxcls) <- names(x)
    names(out$indcls) <- names(x)
    names(out$pval) <- names(x)
    names(out$relfrq) <- levels
    names(out$relabu) <- levels
    names(out$indval) <- levels
    class(out) <- 'indval'
    if (out$error == 1) cat('WARNING: one or more species not assigned to any cluster\n')
    out
}

csi_relabun.ns <- csi_relabun.ns[, colSums(csi_relabun.ns) > 0.05]
bac.ind <- indval(csi_relabun.ns, treatments1)
levels(treatments1)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- csi.tax[which(as.character(csi.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(treatments1)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators.txt",
            sep="\t", row.names = F, quote = F)

split(indicator.bac$Phylum, indicator.bac$Cluster)
split(indicator.bac$Class, indicator.bac$Cluster)

plot(as.factor(indicator.bac$Cluster), as.factor(indicator.bac$Phylum))
```
## AP updates here - cleaned up PCoA ggplots code
# Mixed Effects Models- Phylogenetic Diversity
```{r mixed effects, include=FALSE}
library("lmerTest")
library("pbkrtest")
csi.phylo.ns.45 <- subset(csi.phylo.ns, Date2=="45")
PD.lm.45 <- lmer(PD ~ Salinity*Dispersal + (1|Replicate), data = csi.phylo.ns.45)
plot(PD.lm.45)
summary(PD.lm.45, ddf="Kenward-Roger")
confint(PD.lm.45)
```

```{r (Phylogenetic Diversity - plot), echo=TRUE}
# run full parametric statistical model
PD.lm <- lm(PD ~ Dispersal*Salinity*Date, data = csi.phylo.ns)
plot(PD.lm)
anova(PD.lm)
summary(PD.lm)

PD.lm.45 <- lm(PD ~ Dispersal*Salinity, data = csi.phylo.ns.45)
anova(PD.lm.45)
summary(PD.lm.45)

# run linear regression measured salinity used instead of 'factor' salinity Day 45
PD.reg <- lm(PD~Salinity_real, data = csi.phylo.ns.45)
summary(PD.reg)
```
#Rain Clouds Graphs
```{r rain clouds, include=FALSE}
#Rainclouds for repeated measures
source("../bin/R_rainclouds.R")
source("../bin/summarySE.R")
library(ggpubr)
library(car)
#change Date2 in original df to 1,2,3 instead of 0,18,45
time.Date2 <- as.numeric(csi.phylo.ns$Date2)
csi.phylo.ns.t <- cbind(csi.phylo.ns,time.Date2)
csi.phylo.ns.t$time.Date2 <- recode(csi.phylo.ns.t$time.Date2,"0=1;45=3;18=2")
csi.phylo.ns.t

group <- as.factor(csi.phylo.ns$Salinity)
time <- csi.phylo.ns.t$time.Date2

#Rainclouds for repeated measures - denitrification
p1 <- ggplot(csi.phylo.ns, aes(x = as.factor(time), y = PD, fill = group), size=16) +
  geom_flat_violin(aes(fill = group),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = as.numeric(time)-.15, y = PD, colour = group),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = as.factor(time), y = PD, fill = group),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") + scale_x_discrete(labels=c("1" = "Day 0", "2" = "Day 18", "3" = "Day 45"))
  #+ scale_fill_manual(name="Site", values=c("darkgreen","cyan"))+ scale_colour_manual(name="Site", values=c("darkgreen","cyan"))
p1

ggsave("../figures/Figure_PD_rainclouds.png", plot=last_plot(), device=NULL, path=NULL, scale=1,  width=6, height=6, dpi=300, limitsize=TRUE)
```
#Scatter Plot Graphs
```{r rain clouds, include=FALSE}
# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+facet_wrap(~Date2)+facet_grid(. ~ Date2)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/phylogenetic_div.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

csi.phylo.ns.day45 <- subset(csi.phylo.ns, Date2 == "45")

PD.lm.45 <- lm(PD ~ Dispersal*Salinity, data = csi.phylo.ns.day45)
anova(PD.lm.45)

# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns.day45, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/phylogenetic_div.45.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(csi.phylo.ns.day45, aes(x=PD, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+geom_point(size=4)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Phylogenetic Diversity", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/Cmin_PD.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```

# Ordination - based on taxonomy [included in Werba et al. (In review) manuscript]
```{r (PCoA based on OTU), echo=FALSE}
require("stats")
require("labdsv")
require("mgcv")
require("cluster")
require("vegan")
require("reshape2")
require("reshape")

all.equal(rownames(design.ns.final), rownames(csi_relabun.ns))
sampleREL.dist1 <- vegdist(csi_relabun.ns, method="bray")
csi.full.ns <- cbind(design.ns.final,csi_relabun.ns)

# Set order of salinity
csi.full.ns$Salinity = factor(csi.full.ns$Salinity, levels=c("0","5","9","13"))
design.ns.final$Salinity = factor(design.ns.final$Salinity, levels=c("0","5","9","13"))

# PERMANOVA
adonis = adonis(csi.full.ns[,-c(1:16)] ~ Date2*Dispersal*Salinity, method = "bray", data = csi.full.ns, perm=1000)
adonis

# Principal Coordinates Analysis - NO SOURCE
CSI_pcoa1 <- cmdscale(sampleREL.dist1, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(CSI_pcoa1$eig[1] / sum(CSI_pcoa1$eig), 3) * 100
explainvar2a <- round(CSI_pcoa1$eig[2] / sum(CSI_pcoa1$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #17.3 now 18.2 (?)
explainvar2a #7.3
```
# PCoA Plot - based on taxonomy [included in Werba et al. (In review) manuscript]
```{r (PCoA), include=FALSE}
pcoa.groups <- paste(design.ns.final$Date, design.ns.final$Salinity, sep = "_")

pcoa.points <- data.frame(CSI_pcoa1$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
salinity.p <- c("0","13","5","9","0","13","5","9","0","13","5","9")
date.p <- c("Day 0","Day 0","Day 0","Day 0","Day18","Day18","Day18","Day18","Day 45","Day 45","Day 45","Day 45")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$salinity <- as.factor(salinity.p)
pcoa.cent.dataframe.trts$date <- as.factor(date.p)
pcoa.cent.dataframe.trts$salinity = factor(pcoa.cent.dataframe.trts$salinity, levels=c("0","5","9","13"))
```
# Plot (ggplot)
```{r (Plot PCoA no source), echo=FALSE}
df1a <- as.data.frame(pcoa.cent.dataframe.trts)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=salinity, shape = date,
                 group = interaction(salinity,date))) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=salinity), colour = "black", size=6, stroke = 1) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) + 
  theme(axis.title = element_text(size=20), axis.text=element_text(size=18), 
          axis.text.x = element_text(size=18), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (17.3%)") + ylab("PCoA 2 (7.3%)") + 
  labs(fill = "Salinity", shape = "Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 5, colour="black")),
         shape = guide_legend(override.aes = list(size = 5, fill="black")))

ggsave("../figures/PCoA_SalinityDispersal_NoSource.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=6, height=6, dpi=300, limitsize=TRUE)
```



