---
title: "Salinity affects bacterial phylogeny-function"
author: "Ariane Peralta, Mario Muscarella, Alexandra Stucy, Jo Werba, Mike McCoy"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: default
  fig_caption: yes
  pdf_document: null
editor_options:
  chunk_output_type: console
---
Project Description: Analysis of salinity effects on freshwater bacterial phylogeny-function relationships.

# Setting up working directory, packages
```{r Setup, include=FALSE}
setwd("~/GitHub/CSI_Dispersal/analyses/")
rm(list = ls())

# Set Source R Tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

#Can't figure out how to load phylofactor
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
#BiocManager::install()

#BiocManager::install("ggtree")
#BiocManager::install("Biostrings")
#install.packages('devtools')
#devtools::install_github('reptalex/phylofactor')

# Load Req'd Packages
library("vegan"); library("ecodist"); library("Hmisc")
library("dplyr"); library("reshape2"); library("labdsv")
library("nlme"); library("MASS"); library("ade4"); library("phytools")
library("png"); library("grid"); library("ggplot2")
library("ape"); library("picante"); library("ggtree"); 
library("cowplot"); library("ggpubr"); library("devtools"); library("tidyr")
#library("phylofactor")

source("../bin/R_rainclouds.R")
source("../bin/summarySE.R")
library(ggpubr)
library(cowplot)

# Set Std Err Functions
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

ci <- function(x, ...) {
1.96 * sd(x, na.rm = TRUE)
}

ind.tab <- function(ind = "", pval = 0.05){
  inds <- which(ind$pval <= 0.05)
  indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
  colnames(indicators) <- c("OTU", "Cluster", "IndVal", "Prob")
  indicators$OTU <- names(inds)
  indicators$Cluster <- ind$maxcls[inds]
  indicators$IndVal <- ind$indcls[inds]
  indicators$Prob <- ind$pval[inds]
  return(indicators)
}
```

# Loading data file
```{r Loading Files, include=FALSE}
# NOTES: Tank 3 only had fresh water added 
#        Tank 1 was the source for the salt  
#        Tank 2 was the source for the fresh water
#
# NOTES: Dispersal 0 (treatments 1 and 2) were source tanks-- no dispersal.
#        Dispersal 1 (treatment 3) only received fresh water 
#        Dispersal 3 (treatments 4,6,8,10) only got salt water
#        Dispersal 2 (treatments 5,7,9,11) received both fresh and salt

# Load design file - no source tanks
design.ns <- read.csv("../data/CSI_Design_ENV_NoSourceTanks.csv", row.names=1)
#head(design.ns)
#str(design.ns)
dim(design.ns)

# Load design file - with source tanks but 
#                    missing decomp, nutrients, sreal, Date2 columns
design.full <- read.csv("../data/design_CSI.csv", row.names=1)
design.full <- design.full[-c(grep("mock community", design.full$CSI_ID)), ]
dim(design.full)

# load OTU file
csi_otu <- read.otu("../mothur/CSI.bac.final.shared")
dim(csi_otu)

# load environ/function file - 
# includes expt design, decomp data, Cmin, nutrients day 45 only - need sreal
design.env.full <- read.csv("../data/CSI_Design_ENV_Source.csv", row.names=1)

# Import Taxonomy File for later
csi.tax <- read.tax(taxonomy = "../mothur/CSI.bac.final.taxonomy", 
                    format = "rdp", tax.levels = 6, col.tax = 3)
csi.tax$Phylum[which(csi.tax$Phylum == "Cyanobacteria/Chloroplast")] <- "Cyanobacteria"

# Import Tree File
tree <- read.tree("../mothur/CSI.bac.rename.tree")

# Import UniFrac Distnce Matrix
UniFrac.raw <- read.delim("../mothur/CSI.bac.tree1.weighted.phylip.dist", 
                          skip = 1, header = F, row.names = 1)
row.names(UniFrac.raw) <- gsub("    ", "", row.names(UniFrac.raw))
colnames(UniFrac.raw) <- rownames(UniFrac.raw)
```

## Manipulate data files for statistical analyses and graphing
```{r Bacterial Community Composition, include=FALSE}
# remove extra site (no mock community)

# design only with source tanks ( "CSI041" - not in OTU table )
missing <- setdiff(rownames(design.full), rownames(csi_otu))
design.full <- design.full[-(which(rownames(design.full) == missing)), ]
design.ns <- design.ns[-(which(rownames(design.ns) == missing)), ]
design.env.full <- design.env.full[-(which(rownames(design.env.full) == missing)), ]
dim(design.full)

# OTU table - remove otu's w/ < 2 occurrences across all sites
otu_removal <- csi_otu[, -c(which(colSums(csi_otu) < 2))]
dim(otu_removal)

# Identify low read sites (will remove)
aa <- (rowSums(otu_removal))
aa[which(aa <= 13000)] # CSI033-7180 reads CSI101=85 reads - removed

# Odd sites 
odd.sites <- c("CSI033","CSI101", "CSI130") #IUbarcode #CSI130 is mock community

# Remove odd sitse from OTU table and design files
otu_final <- otu_removal[setdiff(rownames(otu_removal), odd.sites), ]
design_final <- design.full[setdiff(rownames(design.full), odd.sites), ]
design.ns_final <- design.ns[setdiff(rownames(design.ns), odd.sites), ]
design.env.full_final <- design.env.full[setdiff(rownames(design.env.full), odd.sites), ]

all.equal(rownames(design_final), rownames(otu_final))

# create presence/absence and relative abundance matrices
csi_pres_abs <- (otu_final > 0) * 1

csi_relabun <- otu_final
for (i in 1:dim(otu_final)[1]) {
  csi_relabun[i, ] <- otu_final[i, ]/sum(otu_final[i,])
}
```

## Remove Source Tanks
```{r Remove Source Tanks, include=FALSE}
# REMOVE source tanks from otu file
# REMOVE source tanks Number 1,2,3 from otu_final and use design.ns (manually removed source tanks)
tanks.src <- rownames(design_final[which(design_final$Number %in% c(1,2,3)), ])
otu_final.ns <- otu_final[-(which(rownames(otu_final) %in% tanks.src)), ]
otu_final.ns <- otu_final.ns[, -c(which(colSums(otu_final.ns) <= 0))]
dim(otu_final.ns)
dim(design.ns_final)
 
#drop missing data
missing <- setdiff(rownames(design.ns_final), rownames(otu_final.ns))
all.equal(row.names(design.ns_final), rownames(otu_final.ns))

# Drop levels of factors that are no longer in data set 
design.ns.final <- droplevels(design.ns_final)
all.equal(rownames(design.ns.final), rownames(otu_final.ns))

# create presence/absence and relative abundance matrices
csi_pres_abs.ns <- (otu_final.ns > 0) * 1

csi_relabun.ns <- otu_final.ns
for (i in 1:dim(otu_final.ns)[1]) {
  csi_relabun.ns[i, ] <- otu_final.ns[i, ]/sum(otu_final.ns[i, ])
}
```

## Organize Data 
```{r Organize Data, include=FALSE}
# bind design and bact files NO Source
csi.rel.ns <- cbind(design.ns.final,csi_relabun.ns) # rel abundance
csi.pa.ns <- cbind(design.ns.final, csi_pres_abs.ns)
csi.ns <- cbind(design.ns.final,otu_final.ns) # obs abundance
dim(csi.rel.ns)
dim(csi.ns)

# bind design and bact files WITH SOURCE
csi.relabun.full <- cbind(design_final,csi_relabun)
csi.PA.full <- cbind(design_final,csi_pres_abs)
csi.full <- cbind(design_final, otu_final)
dim(csi.relabun.full)
dim(csi.PA.full)

# Set treatments
treatments1 <- as.factor(design.ns.final$Salinity)
levels(treatments1) <- c("0","5","9","13")
treatments2 <- as.factor(design.ns.final$Dispersal)
levels(treatments2) <- c("2","3")

```

# OTU Diversity Metrics [included in Werba et al. 2020 manuscript]
```{r (Diversity Metrics), include=FALSE}
# Rarefy Abundances (min abundance is 13,240. We are sampling to 13,000) for uchime mothur run
# Rarefy Abundances (min abundance is 13,229. We are sampling to 13,000) for vsearch mothur run

min(rowSums(otu_final.ns))
otu.rarefy <- rrarefy(otu_final.ns, 20000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon <- diversity(otu.rarefy, "shannon")

# Species Richness
richness <- rowSums((otu.rarefy >= 1))

# Pielou’s evenness
#J <- shannon/log(specnumber(otu.rarefy[,-c(1:1)])) # something odd here
J <- shannon/log(specnumber(otu.rarefy))


# Combined design,shannon,richness,evenness - no source tanks
csi_otu.div <- cbind(design.ns.final,shannon,richness,J)

# Rarefy Abundances (min abundance is 19846. We are sampling to 19,000) for vsearch mothur run
min(rowSums(otu_final))
otu.rarefy <- rrarefy(otu_final, 19000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon.source <- diversity(otu.rarefy, "shannon")

# Species Richness
richness.source <- rowSums((otu.rarefy >= 1))

# Pielou’s evenness
# J.source <- shannon.source/log(specnumber(otu.rarefy[,-c(1:1)]))
J.source <- shannon.source/log(specnumber(otu.rarefy))


# Combined design,shannon,richness,evenness - no source tanks
csi_otu.div.source <- cbind(design_final, shannon.source, 
                            richness.source,J.source)
```

# Phylogenetic Diversity
```{r (Phylogenetic Diversity), include=FALSE}
str(tree)
dim(otu_final) # Includes source 
dim(csi_otu) # Includes source and odd sites

sum(tree$tip.label %in% colnames(csi_otu) == FALSE) # Tree Represents All

# Small Branches
sum(tree$edge.length < 0.000001) # lots of small branches 

# Unifrac Matrix
dim(UniFrac.raw)

# Subset to remove odd sites (remove source)
UniFrac <- UniFrac.raw[which(row.names(UniFrac.raw) %in%
                                   row.names(otu_final)),
                             which(colnames(UniFrac.raw) %in%
                                   row.names(otu_final))]

dim(UniFrac)

# Make into Distance Matrix
unifrac.dist <- as.dist(UniFrac, upper = T, diag = T)

# Calculate Phylo Diversity (w/ source)
phylo_final <- pd(otu_final, tree, include.root = F)
all.equal(rownames(design_final), rownames(phylo_final))

# Calculate Phylo Diversity (w/o source)
phylo_final.ns <- pd(otu_final.ns, tree, include.root = F)
all.equal(rownames(design.ns.final), rownames(phylo_final.ns))

csi.phylo.ns <- cbind(design.ns.final,phylo_final.ns)

# PERMANOVA UniFrac ??? Fix this Mario 
# adonis = adonis(df[,-c()]~Salinity*Dispersal*Date, method = "bray", data = new.data, perm=1000)
# adonis
```

# Fig 1A: PCoA (UniFrac) - DONE
```{r PCoA UniFrac Ordination, echo=TRUE}
# Principal Coordinates Analysis - INCLUDES SOURCE but no stats
CSI_pcoa_phylo <- cmdscale(unifrac.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1 <- round(CSI_pcoa_phylo$eig[1] / sum(CSI_pcoa_phylo$eig), 3) * 100
explainvar2 <- round(CSI_pcoa_phylo$eig[2] / sum(CSI_pcoa_phylo$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2)

explainvar1 #24.0
explainvar2 #11.8%

all.equal(rownames(design_final), labels(unifrac.dist))

pcoa.groups <- paste(design_final$Date, design_final$Salinity, sep = "_")
pcoa.points <- data.frame(CSI_pcoa_phylo$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)

trt.salt <- c("0","13","5","9","0","13","5","9","0","13","5","9")
trt.date <- c("0","0","0","0","45","45","45","45","18","18","18","18")

pcoa.cent.dataframe.trts$salinity <- as.factor(trt.salt)
pcoa.cent.dataframe.trts$date <- as.factor(trt.date)
pcoa.cent.dataframe.trts$salinity <- factor(c("0","13","5","9"),levels=c("0","5","9","13")) #update order of salinity units

# Plot
df1a <- as.data.frame(pcoa.cent.dataframe.trts)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=salinity, fill=salinity, shape = date)) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=salinity), colour = "black", size=6, stroke = 1) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_fill_manual(values = c("#E69F00", "#56B4E9", "#009E73","#F0E442")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  coord_cartesian(xlim = c(-0.25, 0.25), ylim = c(-0.2, 0.2)) + 
  theme(axis.title = element_text(size=20), axis.text=element_text(size=18), 
          axis.text.x = element_text(size=18), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (24.0%)") + ylab("PCoA 2 (11.8%)") + 
  labs(fill = "Salinity (psu)", shape = "Sampling Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/Phylo_Ordination.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=500, limitsize=TRUE)
```

# Fig 1B: PD Rain Clouds - DONE
```{r PD over salinity rain clouds, echo=TRUE}
# Rainclouds DOES NOT INCLUDE SOURCE

# change Date2 in original df to 1,2,3 instead of 0,18,45
time.Date2 <- as.numeric(csi.phylo.ns$Date2)
csi.phylo.ns.t <- cbind(csi.phylo.ns,time.Date2)
csi.phylo.ns.t$time.Date2 <- recode_factor(csi.phylo.ns.t$time.Date2,
                                           '0'="1", '18'="2", '45'="3")
levels(csi.phylo.ns.t$time.Date2)
csi.phylo.ns.t$Salinity <- as.factor(csi.phylo.ns.t$Salinity)
levels(csi.phylo.ns.t$Salinity)

group <- as.factor(csi.phylo.ns.t$Salinity)
time <- as.factor(csi.phylo.ns.t$time.Date2)

# Plot 
p1 <- ggplot(csi.phylo.ns, aes(x = time, y = PD, fill = group), size=16) + theme_bw() 
p2 <-p1 + geom_flat_violin(aes(fill = group),position = position_nudge(x = .1, y = 0),
                           adjust = 1.5, trim = FALSE, alpha = .5, colour = NA) +
  geom_point(aes(x = as.numeric(time)-.15, y = PD, colour = group), position = position_jitter(width = .05), size = 1.5, shape = 20) +
  geom_boxplot(aes(x = as.factor(time), y = PD, fill = group), outlier.shape = NA, alpha = .5, width = .1, colour = "black") + 
  scale_fill_manual(name="Salinity (psu)", values = c("#E69F00", "#56B4E9", "#009E73","#F0E442")) + 
  scale_color_manual(name="Salinity (psu)", values = c("#E69F00", "#56B4E9", "#009E73","#F0E442")) + 
  labs(x = "Sampling Date", y = "Phylogenetic Diversity") + 
  scale_x_discrete(labels=c("1" = "Day 0", "2" = "Day 18", "3" = "Day 45")) + theme_cowplot()
p2

ggsave("../figures/Figure_PD_rainclouds.png", plot=last_plot(), device=NULL, path=NULL, scale=1,  width=7, height=5, dpi=500, limitsize=TRUE)
```

# Fig 2A: PhyloTree with Indicator Taxa
```{r Phylogenetic Tree with Indicator OTUs, echo=TRUE}
str(tree)

# otu_final, design_final
design.45 <- design_final[which(design_final$Date2 == "45"), ]
design.45 <- design.45[which(design.45$Dispersal %in% c(2, 3)), ]
otus.45 <- otu_final[which(rownames(otu_final) %in% rownames(design.45)), ]
otus.45 <- otus.45[, which(colSums(otus.45) > 100)]

low.phy <- names(which(table(csi.tax$Phylum) < 20))
low.cla <- names(which(table(csi.tax$Class) < 20))

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in%
                                csi.tax$OTU[csi.tax$Phylum %in%
                                c(low.phy)]))]

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in%
                                csi.tax$OTU[csi.tax$Class %in%
                                c(low.cla)]))]

otus.45 <- otus.45[, -c(which(colnames(otus.45) %in% 
                                csi.tax$OTU[csi.tax$Phylum %in%
                                c("Acidobacteria", "Bacteria_unclassified",
                                  "OD1", "Parcubacteria", "Ignavibacteriae",
                                  "BRC1", "Spirochaetes", "Armatimonadetes", 
                                  "Tenericutes", "Deferribacteres",
                                  "Chlamydiae", "Gemmatimonadetes", 
                                  "WS3", "Firmicutes",
                                  "Chloroflexi", "Deinococcus-Thermus")]))]
otus.45 <- otus.45[, -c(which(colnames(otus.45) %in% 
                                csi.tax$OTU[csi.tax$Class %in%
                                c("Bacteroidetes_unclassified",
                           "Bacteroidetes_incertae_sedis_class_incertae_sedis",
                           "Proteobacteria_unclassified", "WS3", "Bacteroidia", "Chlorobi", "Epsilonproteobacteria")]))]

# Rename Error (Fixed in the actual file to prevent errors)
# tree$tip.label <- gsub(".{1}$", "", tree$tip.label)

tree.labs <- tree$tip.label
tree.labs %in% colnames(otus.45)

rm.otus <- setdiff(tree.labs, colnames(otus.45))

tree.2 <- drop.tip(tree, rm.otus)

str(tree.2)

length(tree.2$tip.label)
sum(tree.2$tip.label %in% colnames(otus.45))

tree.otus <- tree.2$tip.label
tree.tax <- csi.tax[which(csi.tax$OTU %in% tree.otus), ]
table(tree.tax$Phylum)

tree.tax.2 <- tree.tax[order(match(tree.tax$OTU, tree.2$tip.label)), ]
all.equal(as.character(tree.tax.2$OTU), tree.2$tip.label)

phylum2 <- tree.tax.2$Phylum
for(i in 1:length(phylum2)){
  if (phylum2[i] == "Proteobacteria"){
    phylum2[i] <- tree.tax.2$Class[i]
  }
}
#   if (phylum2[i] == "Bacteroidetes"){
#     phylum2[i] <- tree.tax.2$Class[i]
#   }
# }

table(phylum2)

groupInfo <- split(tree.2$tip.label, phylum2)
Phy.tree <- groupOTU(tree.2, groupInfo, group_name = "phylum2")
levels(attributes(Phy.tree)$phylum2) <- c(names(groupInfo))

p <- ggtree(Phy.tree, layout="rectangular") + theme(legend.position="top", 
        legend.key = element_rect(colour = NA)) + 
  geom_tippoint(aes(color=phylum2)) 


# Set treatments
treatments1 <- as.factor(design.45$Salinity) 
levels(treatments1) <- c("0","5","9","13")
treatments2 <- as.factor(design.45$Salinity) 
levels(treatments2) <- c("0","1","1","1")
treatments3 <- droplevels(as.factor(design.45$Dispersal))
levels(treatments3) <- c("2", "3")

# Main Indicators
sal.ind <- indval(otus.45, treatments1)
levels(treatments1)
summary(sal.ind)

# Fresh v Salty Indicators
sal.ind2 <- indval(otus.45, treatments2)
levels(treatments2)
summary(sal.ind2)

# Dispersal Indicators
dis.ind <- indval(otus.45, treatments3)
levels(treatments3)
summary(dis.ind)

# Organize Indicator Tables
sal.inds  <- ind.tab(ind = sal.ind, pval = 0.05)
sal.inds2 <- ind.tab(ind = sal.ind2, pval = 0.05) 
dis.inds <- ind.tab(ind = dis.ind, pval = 0.05)  

# Combine for plot
sal.dat <- as.data.frame(matrix(NA, ncol = 5, nrow = length(tree.2$tip.label)))
rownames(sal.dat) <- tree.2$tip.label
colnames(sal.dat) <- c(levels(treatments1), ">0")

for(i in 1:dim(sal.inds)[1]){
  temp <- sal.inds[i, ]
  sal.dat[which(row.names(sal.dat) == temp$OTU), temp$Cluster] <- 10
}

for(i in 1:dim(sal.inds2)[1]){
  temp <- sal.inds2[i, ]
  if(temp$Cluster == 2){
    sal.dat[which(row.names(sal.dat) == temp$OTU), 5] <- 10
  }
}

#csi.tax[which(csi.tax$OTU %in% dis.inds$OTU), ]
#csi.tax[which(csi.tax$OTU %in% sal.inds$OTU), ]
#csi.tax[which(csi.tax$OTU %in% sal.inds2$OTU), ]

# Add heatmap to plot
gheatmap(p, sal.dat, width = 0.5, colnames_angle=90, hjust=1, 
         low = "black", high = "black") + 
  theme(legend.position = "right") + 
  guides(fill = FALSE)   

ggsave("../figures/indicator.tree.png", width = 10, height = 8, units = "in")

```

# Fig. 2B - Indicator Euler Plot
```{r Euler Plot, echo=TRUE}
# Indicator Lists
setdiff(sal.inds$OTU[which(sal.inds$Cluster == 1)], 
        sal.inds2$OTU[which(sal.inds2$Cluster == 1)])

setdiff(sal.inds2$OTU[which(sal.inds2$Cluster == 1)], 
        sal.inds$OTU[which(sal.inds$Cluster == 1)])

inds <- list("0 psu" = sal.inds$OTU[which(sal.inds$Cluster == 1)],
             "5 psu" = sal.inds$OTU[which(sal.inds$Cluster == 2)],
             "9 psu" = sal.inds$OTU[which(sal.inds$Cluster == 3)],
             "13 psu" = sal.inds$OTU[which(sal.inds$Cluster == 4)],
             "0* psu" = sal.inds2$OTU[which(sal.inds2$Cluster == 1)],
             ">0 psu" = sal.inds2$OTU[which(sal.inds2$Cluster == 2)])

library(eulerr)
png("../figures/euler.png",width = 1000, height = 1000, res=200)

set.seed(7)
p1 <- plot(euler(inds, shape = "ellipse"), quantities = TRUE, adjust_labels = T)
p1

dev.off()

```

# ConsenTrait Analysis
```{r ConcenTrait Analysis, include=FALSE}
source("../bin/consenTRAIT.R")
require("phytools")
# Base Tree
str(tree)

# OTUs (abund > 200 obs)
otus.45 <- otu_final[which(rownames(otu_final) %in% rownames(design.45)), ]
otus.45 <- otus.45[, which(colSums(otus.45) > 200)]

# Prune Tree based on max abundance
tree.45 <- drop.tip(tree, setdiff(tree$tip.label, colnames(otus.45)),
                    trim.internal = T)

sum(tree.45$edge.length < 0.000001) # lots of small branches 
tree.45$edge.length[which(tree.45$edge.length < 0.0001)] <- 0.0001 + 
  tree.45$edge.length[which(tree.45$edge.length < 0.0001)] 

# Root tree
tree.45.mid <- midpoint.root(tree.45) #ERROR could not find function "midpoint.root"
tree.45.mid$node.label <- makeNodeLabel(tree.45.mid)$node.label

# Renumber nodes
# n.start <- length(tree.45.mid$tip.label) + 1
# n.end <- n.start + tree.45.mid$Nnode
# tree.45.mid$node.label <- seq(n.start, n.end)
# tree.45.mid$node.label[1:5]
# tree.45.mid <- ape::root(tree.45.mid, resolve.root)

is.rooted(tree.45.mid)

# Tip Data: 50% occupancy
str(otus.45)
design.45$Salinity

dim(otus.45); length(design.45$Salinity)
unique(design.45$Salinity)

sal.trt <- list("0 psu" = c(0), "5 psu" = c(5), "9 psu" = c(9), 
                "13 psu" = c(13), ">0 psu" = c(5, 9, 13))

occupy.45 <- as.data.frame(matrix(NA, nrow = dim(otus.45)[2], 
                                  ncol = length(sal.trt)))
colnames(occupy.45) <- names(sal.trt)
rownames(occupy.45) <- colnames(otus.45)

for(i in 1:length(sal.trt)){
  sites.tmp <- rownames(design.45)[which(design.45$Salinity %in% sal.trt[[i]])]
  otu.tmp <- otus.45[which(rownames(otus.45) %in% sites.tmp), ]
  otu.tmp.pa <- (otu.tmp > 0) * 1
  occupy.tmp <- colMeans(otu.tmp.pa)
  occupy.45[, i] <- (occupy.tmp >= 0.5) * 1
}


str(inds); length(inds)
taxa <- tree.45.mid$tip.label
tip.data <- as.data.frame(matrix(0, nrow = length(taxa), ncol = length(inds)))
colnames(tip.data) <- names(inds)
rownames(tip.data) <- taxa
for(i in 1:length(inds)){
  tmp.data <- inds[[i]]
  for(j in 1:length(tmp.data)){
    tip.data[which(row.names(tip.data) == tmp.data[j]), i] <- 1
  }
}

csi.ct <- consenTRAIT(data = occupy.45, phy = tree.45.mid, cutoff = 0.80)

```

## Fig. 3 - ConcenTrait Plot
```{r ConcentTrait, echo=TRUE}

png("../figures/ConcenTrait.png",width = 1000, height = 1000, res=200)

par(mar = c(5, 7, 1, 1))

plot(x = csi.ct$Distance.L, y = c(1:5),
     xlim = c(-3, 1), ylim = c(0.5, 5.5), cex = 0.1, 
     axes = F, xlab = "", ylab = "")

arrows(x0 = csi.ct$Distance.L, x1 = csi.ct$Distance.L + csi.ct$CI_95.L,
       y0 = 1:5, y1 = 1:5, angle = 90, length = 0.01)

arrows(x0 = csi.ct$Distance.L, x1 = csi.ct$Distance.L - csi.ct$CI_95.L,
       y0 = 1:5, y1 = 1:5, angle = 90, length = 0.01)

points(x = csi.ct$Distance.L, y = c(1:5),
     xlim = c(-2.5, 1), ylim = c(0.5, 5.5),
     pch = 22, bg = "gray", cex = 2)

axis(side = 1, at = c(-3, -2, -1, 0, 1),
     labels = c(0.001, 0.01, 0.1, 1, 10))

axis(side = 2, at = c(1:5),
     labels = csi.ct$Trait, las = 1)

mtext("Average Cluster Depth", side = 1, line = 3, cex = 1.25)
mtext("Grouping", side = 2, line = 5, cex = 1.25)

box(lwd = 1.5)

dev.off()
```

# Fig 4: Phylo Function and Indic Function Relationships
```{r Phylogenetic Fxn and Indic Diversity Fxn, echo=TRUE}
length(inds)

odd.two <- intersect(inds[[2]], inds[[5]])
sal.inds <- setdiff(union(inds[[2]], union(inds[[3]], 
                     union(inds[[4]], inds[[6]]))), odd.two)

OTUs45.rel <- otu_final[which(rownames(otu_final) %in% rownames(design.45)), ]
for (i in 1:dim(OTUs45.rel)[1]) {
  OTUs45.rel[i, ] <- OTUs45.rel[i, ]/sum(OTUs45.rel[i, ] )
}

#Cmin General

#Mass Loss for Each

# Goal: Two panel plot with phylo indicators and phylofactor indicators vs mass loss or cmin. 
# OTU Table
min(rowSums(otu_final.ns))
OTUtable.rare <- rrarefy(otu_final.ns, 20000)
dim(OTUtable.rare)
OTUtable.rare[1:5, 1:5]
OTUtable <- otu_final.ns # t(otu.rarefy)
OTUtable[1:5, 1:5]
OTUtable.rel <- OTUtable
for (i in 1:dim(OTUtable.rel)[1]) {
  OTUtable.rel[i, ] <- OTUtable.rel[i, ]/sum(OTUtable.rel[i, ])
}
dim(OTUtable.rel)
rowSums(OTUtable.rel)

# Indicator Tables
Ind.OTU <- OTUtable[, which(colnames(OTUtable) %in% sal.inds)]
Ind.OTU.rare <- OTUtable.rare[, which(colnames(OTUtable.rare) %in% sal.inds)]
Ind.OTU.rel <- OTUtable.rel[, which(colnames(OTUtable.rel) %in% sal.inds)]
Ind.TREE <- ape::drop.tip(tree.2,setdiff(tree.2$tip.label,colnames(OTUtable.rel)))

Indicator.rel <- rowSums(Ind.OTU.rel)
Indicator.S <- rowSums((Ind.OTU.rare >= 1))
Indicator.H <- diversity(Ind.OTU.rel, "shannon")
Indicator.PD <- pd(Ind.OTU, Ind.TREE, include.root = F)

all.equal(names(Indicator.rel), row.names(csi_otu.div))
all.equal(names(Indicator.S), row.names(csi_otu.div))
all.equal(names(Indicator.H), row.names(csi_otu.div))

csi_otu.div$I_rel <- Indicator.rel
csi_otu.div$I_S <- Indicator.S
csi_otu.div$I_H <- Indicator.H
csi_otu.div$I_PD <- Indicator.PD[, 1]

# Subset only day 45
csi_otu.div.day45 <- subset(csi_otu.div, Date2 == "45")

# Shannon Diversity x Cmin
p <- ggplot(csi_otu.div.day45, aes(x=I_H, y=Cmin, fill = as.factor(Salinity), shape = as.factor(Dispersal))) + 
  geom_point(size=5) + 
  scale_fill_manual(name="Salinity (psu)", values=c("#E69F00", "#56B4E9", "#009E73","#F0E442")) +
  scale_shape_manual(name="Dispersal Treatment", values=c(21,24), labels = c("fresh+salt (2)", "salt only (3)"))
p + theme_bw() + theme(panel.grid.major = element_blank(), 
                        panel.grid.minor = element_blank(), 
                        axis.line = element_line(colour = "black")) + 
  theme(axis.title=element_text(vjust=1,size=16,face="bold"), 
        axis.text = element_text(size=16), 
        axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), 
        panel.border = element_rect(colour = "black",size=1)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  labs(x = "Shannon Diversity Index (H')", 
       y = (expression(paste("C Mineralization (ng CO"[2],
                             " mL"^-{1},"day"^-{1},")")))) + 
  theme(strip.text.x = element_text(size=16, face="bold"), 
        strip.text.y = element_text(size=16, face="bold"), 
        strip.background = element_rect(colour="black", 
                                        fill="white", size=1)) +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/Cmin_shannon.png", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)

# Phylogenetic Diversity x Cmin
p <- ggplot(csi_otu.div.day45, aes(x=I_PD, y=Cmin, fill = as.factor(Salinity), shape = as.factor(Dispersal))) + 
  geom_point(size=5) + 
  scale_fill_manual(name="Salinity (psu)", values=c("#E69F00", "#56B4E9", "#009E73","#F0E442")) +
  scale_shape_manual(name="Dispersal Treatment", values=c(21,24), labels = c("fresh+salt (2)", "salt only (3)"))
p + theme_bw() + theme(panel.grid.major = element_blank(), 
                        panel.grid.minor = element_blank(), 
                        axis.line = element_line(colour = "black")) + 
  theme(axis.title=element_text(vjust=1,size=16,face="bold"), 
        axis.text = element_text(size=16), 
        axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), 
        panel.border = element_rect(colour = "black",size=1)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  labs(x = "Phylogenetic Diversity", 
       y = (expression(paste("C Mineralization (ng CO"[2],
                             " mL"^-{1},"day"^-{1},")")))) + 
  theme(strip.text.x = element_text(size=16, face="bold"), 
        strip.text.y = element_text(size=16, face="bold"), 
        strip.background = element_rect(colour="black", 
                                        fill="white", size=1)) +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/Cmin_PD.png", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)

# Relative Abundance x Cmin
p <- ggplot(csi_otu.div.day45, aes(x=I_rel, y=Cmin, fill = as.factor(Salinity), shape = as.factor(Dispersal))) + 
  geom_point(size=5) + 
  scale_fill_manual(name="Salinity (psu)", values=c("#E69F00", "#56B4E9", "#009E73","#F0E442")) +
  scale_shape_manual(name="Dispersal Treatment", values=c(21,24), labels = c("fresh+salt (2)", "salt only (3)"))
p + theme_bw() + theme(panel.grid.major = element_blank(), 
                        panel.grid.minor = element_blank(), 
                        axis.line = element_line(colour = "black")) + 
  theme(axis.title=element_text(vjust=1,size=16,face="bold"), 
        axis.text = element_text(size=16), 
        axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), 
        panel.border = element_rect(colour = "black",size=1)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  labs(x = "Relative Abundance (Indicators)", 
       y = (expression(paste("C Mineralization (ng CO"[2],
                             " mL"^-{1},"day"^-{1},")")))) + 
  theme(strip.text.x = element_text(size=16, face="bold"), 
        strip.text.y = element_text(size=16, face="bold"), 
        strip.background = element_rect(colour="black", 
                                        fill="white", size=1)) +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/Cmin_RelAbund.png", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)

# OTU richness (S) x Cmin
p <- ggplot(csi_otu.div.day45, aes(x=I_S, y=Cmin, fill = as.factor(Salinity), shape = as.factor(Dispersal))) + 
  geom_point(size=5) + 
  scale_fill_manual(name="Salinity (psu)", values=c("#E69F00", "#56B4E9", "#009E73","#F0E442")) +
  scale_shape_manual(name="Dispersal Treatment", values=c(21,24), labels = c("fresh+salt (2)", "salt only (3)"))
p + theme_bw() + theme(panel.grid.major = element_blank(), 
                        panel.grid.minor = element_blank(), 
                        axis.line = element_line(colour = "black")) + 
  theme(axis.title=element_text(vjust=1,size=16,face="bold"), 
        axis.text = element_text(size=16), 
        axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), 
        panel.border = element_rect(colour = "black",size=1)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  labs(x = "OTU richness", 
       y = (expression(paste("C Mineralization (ng CO"[2],
                             " mL"^-{1},"day"^-{1},")")))) + 
  theme(strip.text.x = element_text(size=16, face="bold"), 
        strip.text.y = element_text(size=16, face="bold"), 
        strip.background = element_rect(colour="black", 
                                        fill="white", size=1)) +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/Cmin_S.png", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

## Mixed Effects Models- Phylogenetic Diversity
```{r diversity fxn mixed effects, echo=TRUE}
library("lmerTest")
library("pbkrtest")
csi.phylo.ns.45 <- subset(csi.phylo.ns, Date2=="45")
PD.lm.45 <- lmer(PD ~ Salinity*Dispersal + (1|Replicate), data = csi.phylo.ns.45)
plot(PD.lm.45)
summary(PD.lm.45, ddf="Kenward-Roger")
confint(PD.lm.45)
```

## PD
```{r (PD diversity xxn), echo=TRUE}
# run full parametric statistical model
PD.lm <- lm(PD ~ Dispersal*Salinity*Date, data = csi.phylo.ns)
plot(PD.lm)
anova(PD.lm)
summary(PD.lm)

PD.lm.45 <- lm(PD ~ Dispersal*Salinity, data = csi.phylo.ns.45)
anova(PD.lm.45)
summary(PD.lm.45)

# run linear regression measured salinity used instead of 'factor' salinity Day 45
PD.reg <- lm(PD~Salinity_real, data = csi.phylo.ns.45)
summary(PD.reg)
```

## OTU PCoA - Bray Curtis (Supp) - DONE
```{r Bray Curtis Ordination supplemental, echo=TRUE}
# Bray Curtis Distance
# cbind(design.ns.final,csi_relabun.ns
sampleREL.dist <- vegdist(csi_relabun.ns, method="bray")

# Principal Coordinates Analysis - NO SOURCE
CSI_PCoA_BC <- cmdscale(sampleREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1 <- round(CSI_PCoA_BC$eig[1] / sum(CSI_PCoA_BC$eig), 3) * 100
explainvar2 <- round(CSI_PCoA_BC$eig[2] / sum(CSI_PCoA_BC$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2)

explainvar1 #17.5
explainvar2 #7.4

pcoa.groups <- paste(design.ns.final$Date, design.ns.final$Salinity, sep = "_")
pcoa.points <- data.frame(CSI_PCoA_BC$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe2 <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe2) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats2 <- rownames(pcoa.cent.dataframe2)
pcoa.cent.dataframe.trts2 <- as.data.frame(pcoa.cent.dataframe2)

trt.salt <- c("0","13","5","9","0","13","5","9","0","13","5","9")
trt.date <- c("0","0","0","0","18","18","18","18","45","45","45","45")

pcoa.cent.dataframe.trts2$salinity <- as.factor(trt.salt)
pcoa.cent.dataframe.trts2$date <- as.factor(trt.date)
pcoa.cent.dataframe.trts2$salinity <- factor(c("0","13","5","9"),levels=c("0","5","9","13")) #update order of salinity units

# Plot
df2a <- as.data.frame(pcoa.cent.dataframe.trts2)
plot1a <- ggplot(df2a, aes(x=V1, y=V2, fill=salinity, shape = date)) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=salinity), colour = "black", size=6, stroke = 1) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_fill_manual(values = c("#E69F00", "#56B4E9", "#009E73","#F0E442")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  #coord_cartesian(xlim = c(-0.25, 0.25), ylim = c(-0.2, 0.2)) + 
  theme(axis.title = element_text(size=20), axis.text=element_text(size=18), 
          axis.text.x = element_text(size=18), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (17.5%)") + ylab("PCoA 2 (7.4%)") + 
  labs(fill = "Salinity (psu)", shape = "Sampling Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/Supp_BrayCurtis_Ordination.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=500, limitsize=TRUE)
```

# PhyloTree with PhyloFactor - not for manuscript
```{r Phylofactor, eval=FALSE, include=FALSE}
# Organize Data

# OTU Table w/ Only "Common Taxa"
OTUtable <- t(otu_final.ns)
OTUtable[1:5, 1:5]
for (i in 1:dim(OTUtable)[2]) {
  OTUtable[, i] <- OTUtable[, i]/sum(OTUtable[, i])
}
dim(OTUtable)
colSums(OTUtable)
OTUtable <- OTUtable[which(colSums(otu_final.ns) > 500), ]
OTUtable <- OTUtable[, which(rowSums(OTUtable) > 0)]
dim(OTUtable)
colSums(OTUtable) # removed about 10%
OTUtable[which(OTUtable == 0)] <- 0.0000001
dim(OTUtable)
colSums(OTUtable)

# Independent Variables: Salinity
all.equal(colnames(OTUtable), rownames(csi.ns))
trt <- as.factor(csi.ns$Salinity)
levels(trt) <- c("0","5","9","13")

# Phylogenetic Tree
tree.phy <- ape::drop.tip(tree,setdiff(tree$tip.label,rownames(OTUtable)))
OTUtable <- OTUtable[tree.phy$tip.label, ]
dim(OTUtable)
colSums(OTUtable)

# Taxonomy
csi.tax.common <- csi.tax[which(csi.tax$OTU %in% rownames(OTUtable)), ]
taxonomy <- tax.merge(csi.tax.common)

test <- ses.mntd(OTUtable, )


# PhyloFactorization 4 Varaibles
PF <- PhyloFactor(OTUtable, tree.phy, trt, nfactors = 10) # > 500 factors
PF$factors
dim(OTUtable)

# Phylofactoraization with Saline vs Non-Saline
trt2 <- as.factor(csi.ns$Salinity)
levels(trt2) <- c("0","1","1","1")
PF2 <- PhyloFactor(OTUtable, tree.phy, trt2, nfactors = 10)
PF2$factors

dim(OTUtable)
s <- pf.summary(PF2,taxonomy,factor=1) 
s$model
td <- pf.tidy(s) 


# Summary
# A tidier summary tool
s <- pf.summary(PF,taxonomy,factor=1) 
s$model
td <- pf.tidy(s)                                 
td$`group1, Monophyletic`                          
# Simplified group IDs - the unique shortest unique prefixes separating the groups
td$`group2, Monophyletic`

## Plotting with summary tools ##
par(mfrow=c(1,1))
plot(as.numeric(trt),td$`Observed Ratio of group1/group2 geometric means`,
       ylab='Average ratio of Group1/Group2',pch=18,cex=2)
lines(td$`Predicted ratio of group1/group2`,lwd=2)
legend(1,12,legend=c('Observed','Predicted'),pch=c(18,NA),lwd=c(NA,2),
       lty=c(NA,1),cex=2)

# A tidier summary tool
s <- pf.summary(PF,taxonomy,factor=2) 
td <- pf.tidy(s)                                 
td$`group1, Monophyletic`                          
# Simplified group IDs - the unique shortest unique prefixes separating the groups
td$`group2, Paraphyletic`

```

## PD Scatter Plot Graphs - All OTUs
```{r Scatter Plots Div Fxn All OTUs, include=FALSE}
# Phylogenetic Diversity - Treatment Salinity

# Graphing Shannon Diversity
date.labs <- c("Day 0", "Day 18", "Day 45") #for facet labels
names(date.labs) <- c("0", "18", "45")

p <- ggplot(csi.phylo.ns, aes(x=as.factor(Salinity), y=PD, color=as.factor(Dispersal))) + 
  geom_boxplot() +
  geom_point(aes(color=as.factor(Dispersal)), size=1, position = position_jitterdodge()) +       
  scale_color_manual(name="Dispersal Treatment", values=c("black", "gray"), labels = c("fresh+salt (2)", "salt only (3)"))  +
  facet_wrap(~Date2) + facet_grid(. ~ Date2)
p1<-p+facet_wrap(~Date2)+facet_grid(. ~ Date2,labeller = labeller(Date2=date.labs)) 
# + theme(strip.text = element_text(face = "bold"))
p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity (psu)", y = "Phylogenetic Diversity (PD)") + 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1)) 

ggsave("../figures/PD_disp_salinity_time.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=500, limitsize=TRUE)


csi.phylo.ns.day45 <- subset(csi.phylo.ns, Date2 == "45")

PD.lm.45 <- lm(PD ~ Dispersal*Salinity, data = csi.phylo.ns.day45)
anova(PD.lm.45)

# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns.day45, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

#ggsave("../figures/phylogenetic_div.45.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(csi.phylo.ns.day45, aes(x=PD, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+geom_point(size=4)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Phylogenetic Diversity", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

#ggsave("../figures/Cmin_PD.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```

## Phylogenetic Linear Models (Supp) - All OTUs
```{r (Phylogenetic Diversity all OTUs), eval=FALSE, include=FALSE}
# run full parametric statistical model
PD.lm <- lm(PD ~ Dispersal*Salinity*Date, data = csi.phylo.ns)
plot(PD.lm)
anova(PD.lm)

# run linear regression measured salinity used instead of 'factor' salinity
PD.reg <- lm(PD~Salinity_real, data = csi.phylo.ns)
summary(PD.reg)

# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+facet_wrap(~Date2)+facet_grid(. ~ Date2)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/phylogenetic_div.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

csi.phylo.ns.day45 <- subset(csi.phylo.ns, Date2 == "45")

PD.lm.45 <- lm(PD ~ Dispersal*Salinity, data = csi.phylo.ns.day45)
anova(PD.lm.45)

# Phylogenetic Diversity - Treatment Salinity
p <- ggplot(csi.phylo.ns.day45, aes(x=Salinity, y=PD, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Phylogenetic Diversity (PD)") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/phylogenetic_div.45.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(csi.phylo.ns.day45, aes(x=PD, y=Cmin, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","gray"), labels = c("fresh+salt (2)", "salt only (3)")) + stat_summary(fun.data=mean_cl_boot,size=1,position=position_dodge(width=2)) 
p1=p+geom_smooth(method="lm", se=FALSE)+geom_point(size=4)
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=16), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=16), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Phylogenetic Diversity", y = (expression(paste("C Mineralization (ng CO"[2]," mL"^-{1}," day"^-{1},")")))) + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/Cmin_PD.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```