---
title: "Bacterioplankton response to salinity (aka 2015 CSI Dispersal Experiment)"
author: "Alex Stucy, Jo Werba, Mike McCoy, Ariane Peralta"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
output: 
  pdf_document:
  fig_caption: true
---
Project Description: Analysis of salinity and dispersal of saline aquatic communities on bacterial community structure and function.

Setting up working directory, packages
```{r (1)}
#PC - set WD manually by Session -> Set Working Directory -> Choose Directory...
rm(list = ls())

#Set source R tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

#load req'd packages
require("vegan")
require("dplyr")
require("nlme")
require("reshape2")
require("ecodist")
require("ggplot2")
require("ade4")
require("png")
require("MASS")
require("grid")
require("ape")
require("png")
require("picante")

# set std err
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}
ci <- function(x, ...) {
1.96 * sd(x, na.rm = TRUE)
}
```

Loading data files and manipulate data files for statistical analyses and graphing
```{r (2)}
# load design file - no source tanks
design.ns <- read.csv("../data/CSI_Design_ENV_NoSourceTanks.csv", row.names=1)
head(design.ns)
str(design.ns)
dim(design.ns)

# load design file - with source tanks but missing decomp, nutrients, sreal, Date2 columns
design.full <- read.csv("../data/design_CSI.csv", row.names=1)
design.full <- design.full[-c(grep("mock community", design.full$CSI_ID)), ]
dim(design.full)

# load OTU file
csi_otu <- read.otu("../data/CSI.shared")
dim(csi_otu)

# remove extra site (no mock community)

# design with source tanks
missing <- setdiff(rownames(design.full), rownames(csi_otu))
design.full <- design.full[-(which(rownames(design.full) == missing)), ]
dim(design.full)

# OTU table - remove otu's w/ < 2 occurrences across all sites
otu_removal <- csi_otu[, which(colSums(csi_otu) >= 2)]
dim(otu_removal)

aa <- (rowSums(otu_removal))
aa # CSI033-7180 reads CSI101=75 reads - removed

# OTU table - removed low abundance samples
csi_low_remov <- otu_removal[which(rowSums(otu_removal) >= 13000), ]
dim(csi_low_remov)

# OTU table - odd sites in bacterial composition data and remove in design file
odd.sites <- c("CSI033","CSI101")

otu_final <- csi_low_remov[setdiff(rownames(csi_low_remov), odd.sites), ]
design_final <- design.full[setdiff(rownames(design.full), odd.sites), ]
design.ns <- design.ns[setdiff(rownames(design.ns), odd.sites), ]

all.equal(rownames(design_final), rownames(otu_final))

# create presence/absence and relative abundance matrices
csi_pres_abs <- (otu_final > 0) * 1
csi_relabun <- otu_final
for (i in 1:dim(otu_final)[1]) {
csi_relabun[i, ] <- otu_final[i, ]/sum(otu_final[i,
])
}

# import taxonomy file for later
csi.tax <- read.tax(taxonomy = "../data/CSI.0.03.cons.taxonomy", format = "rdp", tax.levels = 6, col.tax = 3)

# bind design and bact files
csi.full <- cbind(design_final,csi_relabun)

# REMOVE source tanks from otu file
# REMOVE source tanks Number 1,2,3 from otu_final and use design.ns (manually removed source tanks)
temp <- rownames(design_final[which(design_final$Number %in% c(1,2,3)), ])
otu_final.ns <- otu_final[-(which(rownames(otu_final) %in% temp)), ]
dim(otu_final.ns)
dim(design.ns)

#drop missing data
missing <- setdiff(rownames(design.ns), rownames(otu_final.ns))
design.ns2 <- design.ns[-(which(rownames(design.ns) == missing)), ]
dim(design.ns2)

# Drop levels of factors that are no longer in data set 
design.ns.final <- droplevels(design.ns2)

all.equal(rownames(design.ns2), rownames(otu_final.ns))

# create presence/absence and relative abundance matrices
csi_pres_abs.ns <- (otu_final.ns > 0) * 1
csi_relabun.ns <- otu_final.ns
for (i in 1:dim(otu_final.ns)[1]) {
csi_relabun.ns[i, ] <- otu_final.ns[i, ]/sum(otu_final.ns[i,
])
}
# bind design and bact files
csi.full.ns <- cbind(design.ns.final,csi_relabun.ns)

csi.ns <- cbind(design.ns.final,otu_final.ns)

# Set treatments
treatments1 <- as.factor(design.ns.final$Salinity)
levels(treatments1) <- c("0","5","9","13")
treatments2 <- as.factor(design.ns.final$Dispersal)
levels(treatments2) <- c("2","3")

date_1 <- as.factor(design.ns.final$Date2)
```

Calculating Diversity Metrics
```{r (3)}
# Rarefy Abundances (min abundance is 13,240. We are sampling to 13,000)
min(rowSums(otu_final.ns))
otu.rarefy <- rrarefy(otu_final.ns, 13000)

# Calculate Shannon H' (called shannon) using full data set (WITH source tanks)
shannon <- diversity(otu.rarefy, "shannon")

# Species Richness
richness <- rowSums((otu.rarefy >= 1))

# Pielouâ€™s evenness
J <- shannon/log(specnumber(otu.rarefy[,-c(1:1)]))

# Dombined design,shannon,richness,evenness - no source tanks
csi_otu.div <- cbind(design.ns.final,shannon,richness,J)
```

(1) How does salinity/dispersal/date influence bacterial diversity?
-Calculate Shannon's H' (diversity metric) - no source tanks
-Run stat model diversity~dispersal|salinity|date

Supplemental Results:
Use full data set (with source tanks) #Maybe just subset source tanks into a separate figure with no stats (just use as supplemental)

Testing salinity x dispersal influence on DIVERSITY (no source tanks)
```{r (4)}
# run full parametric statistical model
shannon.lm <- lm(shannon ~ Dispersal*Salinity*Date, data = csi_otu.div)
plot(shannon.lm)
shannon.lm
anova(shannon.lm)

# run linear regression measured salinity used instead of 'factor' salinity
shannon.reg <- lm(shannon~Salinity_real, data = csi_otu.div)
summary(shannon.reg)

```

Plotting diversity metrics
```{r (5)}
date_1 <- as.factor(design.ns.final$Date2)
labels <- c("0"="Day 0","18"="Day 18", "45"="Day 45")

# Graphing Shannon Diversity - Treatment Salinity 
p <- ggplot(csi_otu.div, aes(x=Salinity, y=shannon, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh (2)", "fresh+salt (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Date2)+facet_grid(. ~ Date2, labeller=labeller(Date2 = labels))
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Shannon Diversity Index (H')") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/shannon.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```
(2) How does salinity/dispersal/date influence bacterial evenness?

```{r}
# run full parametric statistical model
J.lm <- lm(J ~ Dispersal*Salinity*Date, data = csi_otu.div)
plot(J.lm)
anova(J.lm)

# run linear regression measured salinity used instead of 'factor' salinity
J.reg <- lm(J~Salinity_real, data = csi_otu.div)
summary(J.reg)

# Graphing Pielous J - Treatment Salinity
p <- ggplot(csi_otu.div, aes(x=Salinity, y=J, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh (2)", "fresh+salt (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Date2)+facet_grid(. ~ Date2, labeller=labeller(Date2 = labels))
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Pielou's Evenness (J')") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))

ggsave("../figures/pielousj.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

```


(3) How does salinity/dispersal/date influence bacterial richness?

```{r}
# run full parametric statistical model
richness.lm <- lm(log(richness) ~ Dispersal*Salinity*Date, data = csi_otu.div)
plot(richness.lm)
richness.lm
anova(richness.lm)

# run linear regression measured salinity used instead of 'factor' salinity
richness.reg <- lm(richness~Salinity_real, data = csi_otu.div)
summary(richness.reg)

# Graphing richness - Treatment Salinity
p <- ggplot(csi_otu.div, aes(x=Salinity, y=richness, color=as.factor(Dispersal)))+ scale_color_manual(name="Dispersal Treatment", values=c("black","purple"), labels = c("fresh (2)", "fresh+salt (3)")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Date2)+facet_grid(. ~ Date2, labeller=labeller(Date2 = labels))
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=16,face="bold"), axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=14), panel.border = element_rect(colour = "black",size=1)) + theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Salinity", y = "Species Richness") + theme(strip.text.x = element_text(size=16, face="bold"), strip.text.y = element_text(size=16, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1))
  
ggsave("../figures/richness.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```


(4) How does salinity and dispersal influence carbon mineralization rates?
Calculate C mineralization rate (should be completed already in CSI shared folder)
Graph C mineralization
Run stat model Cmin~dispersal + salinity + dispersal*salinity

```{r (4 w source tanks)}
#Cmin data inculded in full data set - no longer need to import separate file. Note: only use Date2=45 since Cmin only completed Day 45 of experiment

#subset according to Date2=45 only b/c Cmin only measured for Date2=45
csi.full.ns.Cmin  <- csi.full.ns[which(csi.full.ns$Date2=="45"),]
dim(csi.full.ns)
dim(csi.full.ns.Cmin)

# run full parametric statistical model
Cmin.lm <- lm(Cmin ~ Dispersal*Salinity + Dispersal + Salinity, data = csi.full.ns.Cmin)
Cmin.lm
anova(Cmin.lm)

# run linear regression measured salinity used instead of 'factor' salinity
Cmin.reg <- lm(Cmin~Salinity_real, data = csi.full.ns.Cmin)
summary(Cmin.reg)

#import C-mineralization CSV -- no source tanks
carbon_min <- read.csv("../data/Cmin-data-NO-SOURCE.csv")

p <- ggplot(csi.full.ns.Cmin, aes(x=Salinity, y=Cmin, color=Dispersal))+geom_point()
p1=p+geom_smooth(method="lm",aes(fill=Dispersal))
p1 + theme_bw() 
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=10,face="bold"), axis.text=element_text(size=8), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=8), panel.border = element_rect(colour = "black",size=1.25)) + theme(axis.ticks.length=unit(0.3,"cm")) + xlab("Salinity") + ylab("Carbon Mineralization (ng CO2/mL/day)") 
ggsave("../figures/carbon_mineralization.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(csi.full.ns.Cmin, aes(x=Salinity_real, y=Cmin, color=Dispersal))+geom_point()
p1=p+geom_smooth(method="lm",aes(fill=Dispersal))
p1 + theme_bw() 
p1 + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +theme(axis.title=element_text(vjust=1,size=10,face="bold"), axis.text=element_text(size=8), axis.text.x = element_text(vjust=0.65, hjust=0.5, size=8), panel.border = element_rect(colour = "black",size=1.25)) + theme(axis.ticks.length=unit(0.3,"cm")) + xlab("Measured Salinity") + ylab("Carbon Mineralization (ng CO2/mL/day)") 
ggsave("../figures/real.sal.carbonmin.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```


(6) How does salinity and dispersal influence bacterial community composition? (this should be done - just combine with your Rmd file and double check)
--Complete with source tanks
--Complete without source tanks (SFS poster code)
Revisit original data set and include source tanks to Rmd file - still waiting on full env data from Jo Werba
Graph PCoA ordination (include source tanks) - salinity=colors, date=shapes, label source tanks
Run PERMANOVA to test bacterial community ~ salinity + dispersal + salinity*dispersal

```{r (6 with NO SOURCE tanks)}
require("stats")
require("labdsv")
require("mgcv")
require("cluster")
require("vegan")
require("reshape2")
require("reshape")

#csi.full <- cbind(design_final,csi_relabun) - ready for stats
#csi.full.ns <- cbind(design.ns.final,csi_relabun.ns) - ready for stats NO SOURCE TANKS

sampleREL.dist1 <- vegdist(csi_relabun.ns, method="bray")

# PERMANOVA
adonis = adonis(csi.full.ns[,-c(1:16)] ~ Date2*Dispersal*Salinity, method = "bray", data = csi.full.ns, perm=1000)
adonis

# Principal Coordinates Analysis - NO SOURCE
CSI_pcoa1 <- cmdscale(sampleREL.dist1, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(CSI_pcoa1$eig[1] / sum(CSI_pcoa1$eig), 3) * 100
explainvar2a <- round(CSI_pcoa1$eig[2] / sum(CSI_pcoa1$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #17.2
explainvar2a #7.4
```

#PCoA set up WITH source tanks
```{r}
sampleREL.dist2 <- vegdist(csi_relabun, method="bray")

# PERMANOVA
adonis2 <- adonis(csi.full[,-c(1:16)] ~ Date*Dispersal*Salinity, method = "bray", data = csi.full, perm=1000)
adonis2

# Principal Coordinates Analysis - NO SOURCE
CSI_pcoa2 <- cmdscale(sampleREL.dist2, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(CSI_pcoa2$eig[1] / sum(CSI_pcoa2$eig), 3) * 100
explainvar2b <- round(CSI_pcoa2$eig[2] / sum(CSI_pcoa2$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #20.8
explainvar2b #6.4
```


# PCoA Plot Design without Source
```{r}
all.equal(rownames(design.ns.final), rownames(csi_relabun.ns))

pcoa.groups <- paste(design.ns.final$Date, design.ns.final$Salinity, sep = "_")

pcoa.points <- data.frame(CSI_pcoa1$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Salinity
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Date
```

# Plot (ggplot)
```{r}
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                 group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col), colour = "black", size=6, stroke = 0.75) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(labels = c("0","5","9","13"), 
                      values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_fill_manual(labels = c("0","5","9","13"), 
                    values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  coord_cartesian(xlim = c(-0.25, 0.5), ylim = c(-0.35, 0.3)) + 
  theme(axis.title = element_text(size=18), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (17.2%)") + ylab("PCoA 2 (7.4%)") + 
  labs(fill = "Salinity", shape = "Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/16SrRNA_CSI_Rplot_SalinityDispersal_NoSource.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```

#PCoA plot design WITH SOURCE TANKS - need to update and figure out symbol for SOURCE - McCoy

```{r}
all.equal(rownames(design_final), rownames(csi_relabun))

pcoa.groups2 <- paste(design_final$Date, design_final$Salinity, sep = "_")

pcoa.points2 <- data.frame(CSI_pcoa2$points, group = pcoa.groups2)

# Calculate Centroids (mean and SE)
pcoa.L.centroids2 <- melt(pcoa.points2, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids2 <- acast(pcoa.L.centroids2, variable ~ group, mean)
pcoa.centroids.se2 <- acast(pcoa.L.centroids2, variable ~ group, se)
pcoa.centroids.sd2 <- acast(pcoa.L.centroids2, variable ~ group, sd)

# Combine
pcoa.cent.dataframe2 <- cbind(t(pcoa.centroids2), t(pcoa.centroids.se2))
colnames(pcoa.cent.dataframe2) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats2 <- rownames(pcoa.cent.dataframe2)

pcoa.col2 <- as.factor(sapply(strsplit(pcoa.cent.treats2, "_"), `[`, 2)) # Salinity
pcoa.shape2 <- as.factor(sapply(strsplit(pcoa.cent.treats2, "_"), `[`, 1))  # Date
```

# PLOT GGPLOT WITH SOURCE TANKS (where is salinity 0???) - need to fix figure to show salinity 0 and SOURCES
```{r}
df1a2 <- as.data.frame(pcoa.cent.dataframe2)
plot1a2 <- ggplot(df1a2, aes(x=V1, y=V2, colour=pcoa.col2, shape = pcoa.shape2,
                 group = interaction(pcoa.col2, pcoa.shape2))) + theme_bw() 
plot1a2 + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col2), colour = "black", size=6, stroke = 0.75) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  scale_colour_manual(labels = c("0","5","9","13"), 
                      values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_fill_manual(labels = c("0","5","9","13"), 
                    values = c("#FFFFFF", "#00FFFF", "#33CCCC", "#0066CC")) + 
  scale_shape_manual(labels = c("0","18","45"),
                     values = c(22, 21, 24)) +
  coord_cartesian(xlim = c(-0.25, 0.5), ylim = c(-0.35, 0.3)) + 
  theme(axis.title = element_text(size=18), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (20.8%)") + ylab("PCoA 2 (6.4%)") + 
  labs(fill = "Salinity", shape = "Date") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")),
         shape = guide_legend(override.aes = list(size = 4, fill="black")))

ggsave("../figures/16SrRNA_CSI_Rplot_SalinityDispersal_Source.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```

# Indicator Species Analysis

(5) Which bacterial taxa are driving bacterial community patterns in response to salinity only?
Subset data (run without source tanks)
Bray-Curtis dissimilarity matrix for bacterial community
Run Species Indicator Analysis - see IL_Wetlands.Rmd file
Create Table for Supplemental (includes all taxa and associated taxon contribution to bacterial response)
Create subset in table format to highlight top 5-10 OTUs (with names) for each salinity tank type (or by whatever groupings seem interesting) - See AP's IL Wetlands paper

```{r}
library("labdsv")
#bac.ind

csi_relabun.ns.2 <- csi_relabun.ns[, colSums(csi_relabun.ns) > 0.05]
bac.ind <- indval(csi_relabun.ns.2, treatments1)
levels(treatments1)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- csi.tax[which(as.character(csi.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(treatments1)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators.txt",
            sep="\t", row.names = F, quote = F)

split(indicator.bac$Phylum, indicator.bac$Cluster)
split(indicator.bac$Class, indicator.bac$Cluster)

plot(as.factor(indicator.bac$Cluster), as.factor(indicator.bac$Phylum))

```

(7)  Is there a relationship between community composition and litter decomposition rates? (Date2=45 only)
Subset data - source tanks only
Euclidean distance for leaf litter decomposition
Bray-Curtis dissimilarity matrix for bacterial community
 Run matrix comparison using Mantel test

Subset data -  experimental (all data but source) tanks
Euclidean distance for leaf litter decomposition
Bray-Curtis dissimilarity matrix for bacterial community
Run matrix comparison using Mantel test
Run Redundancy Analysis to include litter decomp AND C min
Model:  bacterial community~litter decomp1+decomp2+decomp3+Cmin

```{r (7 without source tanks)}
#Is there a relationship betwen bacterial community composition and decomposition rate? used matrix comparison Mantel test
str(csi.full.ns.Cmin)
df.bcc <- csi.full.ns.Cmin[,-c(1:16)] #bacteria
df.env.bcc <- csi.full.ns.Cmin[,c(1:16)] #env
df.decomp <- csi.full.ns.Cmin[,c(12:14)] #decomp
df.decompCmin <- csi.full.ns.Cmin[,c(12:15)] #decompCmin

#bray curtis for bact comm
dist.bcc <- vegdist(df.bcc, method = "bray")
#euclidean distance for leaf litter decomp
dist.decomp <- vegdist(df.decomp, method = "euclidean")
#euclidean distance for leaf litter decomp and Cmin
dist.decompCmin <- vegdist(df.decompCmin, method = "euclidean")

#matrix comparison
mantel.rtest(dist.bcc, dist.decomp, nrepet = 999) #bacteria compared to decomp correlation r = 0.231, P=0.006

mantel.rtest(dist.bcc, dist.decompCmin, nrepet = 999) #bacteria compared to decompCmin correlation r = 0.212, P=0.007

#Redundancy Analysis
f <- df.bcc ~ Maple_dmass + Spartina_dmass + Phrag_dmass + Cmin
df.rda <- dbrda(f, data=df.env.bcc, distance="bray")
anova(df.rda, by="terms", model="direct")
RsquareAdj(df.rda)

plot(df.rda)


```

```{r}
# Phylogenetic Diversity
# Load Tree
tree <- read.tree("../data/CSI.bac.rename.tree")
tree$tip.label <- gsub("\\|", "", tree$tip.label)

sum(tree$tip.label %in% colnames(csi_otu) == FALSE)

# Small Branches
sum(tree$edge.length < 0.0000001)

# Import Unifrac Matrix
unifrac.raw <- read.delim("../data/CSI.bac.tree1.weighted.phylip.dist",
                             header = F, skip = 1, row.names = 1)

row.names(unifrac.raw) <- gsub("    ", "", row.names(unifrac.raw))

unifrac <- unifrac.raw[which(row.names(unifrac.raw) %in%
                                   row.names(otu_final)),
                             which(row.names(unifrac.raw) %in%
                                   row.names(otu_final))]

rownames(unifrac) <- gsub("     ", "", row.names(unifrac))
colnames(unifrac) <- rownames(unifrac)
dim(unifrac)

# Make into Distance Matrix
unifrac.dist <- as.dist(unifrac, upper = T, diag = T)

# Calculate Phylo Diversity
phylo.d <- pd(otu_final, tree, include.root = F)
```

```{r}
#tree.2 <- reorder(tree, order = "cladewise")
#tree.r <- midpoint.root(tree.2)
phylo.d <- pd(otus.r, tree, include.root = F)

design.r <- design.otu[which(rownames(design.otu) %in% rownames(otus.r)), ]


design.sed <- design.r[design.r$Sample_Type != "W", ]

sed.bac <- as.data.frame(matrix(NA, dim(design.sed)[1], 10))
colnames(sed.bac) <- c("Sample", "Location", "Time", "Type", "Replicate", 
                       "S", "F_alpha", "S_even", "Shannon", "Phylo")
sed.bac$Sample <- rownames(design.sed)
sed.bac$Location <- design.sed$Location
sed.bac$Time <- design.sed$Storm
sed.bac$Type <- design.sed$Sample_Type
sed.bac$Replicate <- design.sed$Field_replicate
for(i in 1:length(rownames(design.sed))){
  temp <- rownames(design.sed)[i]
  sed.bac$S[i] <- richness[which(names(richness) == temp)]
  sed.bac$F_alpha[i] <- fisher[which(names(fisher) == temp)]
  sed.bac$S_even[i] <- simp.even[which(names(simp.even) == temp)]
  sed.bac$Shannon[i] <- shannon[which(names(shannon) == temp)]
  sed.bac$Phylo[i] <- phylo.d[which(rownames(phylo.d) == temp), 1]
}

# Reorganize
sed.bac.m <- melt(sed.bac, id = c("Sample", "Location", "Time", "Type"))
sed.bac.c <- cast(data = sed.bac.m, Location + Type ~ variable, c(mean, se), na.rm=T)

sed.bac.c <- as.data.frame(sed.bac.c)
sed.bac.m <- as.data.frame(sed.bac.m)

# # ANOVA - update
aov.sed=aov(value~Location*Type,data=sed.bac.m[which(sed.bac.m$variable == "S"), ])
summary(aov.sed)
posthocD <-TukeyHSD(aov.sed,"Location:Type",conf.level=0.95)
posthocD

```
```
